\documentclass[10pt,a4paper,twocolumn,article]{memoir}
\counterwithout{section}{chapter}

\usepackage[T1]{fontenc}
\usepackage{palatino}
\usepackage{graphicx}
  \graphicspath{{./figs/}}

%% remove colorlinks when preparing for print
\usepackage[final,hyperindex,hyperfootnotes,bookmarksnumbered,colorlinks]{hyperref}

\usepackage{eqparbox}
  %% keep size of first column of sequence align table on memory to calculate
  %% the size limit for the second column
  \newsavebox{\firstentrybox}
  \newcolumntype{F}{%
    >{\begin{lrbox}{\firstentrybox}}%
      l%
    <{\end{lrbox}%
    \eqmakebox[firstentry][l]{\unhcopy\firstentrybox}}}

\usepackage{fp}
\usepackage{fmtcount}
\usepackage{seqsplit}
\usepackage{multirow}
\usepackage{url}
\usepackage[textsize=footnotesize]{todonotes}
  %% new command for box about missing references
  \newcommand{\addref}[1][]{\todo[color=red!40,size=\tiny]{Add reference: #1}}

\usepackage{natbib}

%% to keep a pair of related table and figures, we set a new float and use it
%%conjunction with the captionof command from capt-of
\usepackage{capt-of}
\newfloat{TableAndFigure}{X}{capname}

\input{results/variables-align_results}
\input{results/variables-cluster_stats}
\input{results/variables-protein_stats}
\input{data/variables-Marzluff-stats}

\author{David Miguel Susano Pinto and Andrew Flaus}
\title{The canonical core histone catalogue}

\begin{document}
  %% Text flow: first introduction on what are histones and nucleosomes. Some history
  %% on them, and explanation on the different types. Then we list their organization,
  %% following the same pattern of gene expression: first describe genes and their
  %% organization, then their transcripts, and finally the proteins. One everything is
  %% presented, discuss on how the current division of histones is inappropriate and
  %% leads to misconceptions about histone variance. Finalize with comment on reproducible
  %% research and the code availability to rebuild all tables and analysis of this paper,
  %% and how we have used it for other organisms.

  \maketitle

  \begin{abstract}
    The core histone proteins are encoded by a large family of genes distributed in clusters across the
    genome. It is often overlooked that the canonical histone genes display variation within
	themselves, encoding \TotalHTwoA{} H2A, \TotalHTwoB{} H2B, \TotalHThree{} H3, and \TotalHFour{} H4
    unique protein sequences. This canonical histone isoforms diversity exists
	in addition to the more divergent histone variants that perform specific chromatin roles.
	
\todo{say something about nature of canonical variation}
	The most divergent isoforms differ in protein sequence for H2A by ?\%, H2B by ?\%, H3 vy ?\% and H4
	by ?\%. 

    Annotation of the human genome is in flux and is constantly being updated and enriched in reference
	databases such as RefSeq. To avoid the obsolescence of manual surveys but maintain the readability
	of scientific literature, we have developed a novel software framework that dynamically generates
	an up-to-date canonical histone catalogue based on human genome annotations and inserts the
	information into this paper format. We show how this can be readily adapted to histones of other
	organisms and applied to other gene families.
  \end{abstract}

  %%%%%%%%%%%%%%%%
  %%
  %% Introduction
  %%
  %%%%%%%%%%%%%%%%

  \section{Introduction}
    %% Flow of introduction: histones are big part of the chromatin, the building unit.
    %% Then first division, split by type. When we know the type, we can explain
    %% nucleosome structure, which leads to core vs linker division. Then more history,
    %% we found variants. New division of classes, canonical vs variant. Explain the
    %% mess of variants, with all the names used for each division, and sometimes
    %% clashing nomenclature. Finalize with what this paper is about.

    Histones are among the most abundant proteins in eukaryotic cells and contribute up
    to half the mass of chromatin. They define the structure and accessibility of the
    nucleosome as the fundamental repeating unit of genome organisation around which
    the DNA is wrapped upon. In addition, the reactive
    side chains of histones are post-translationally modified as a nexus for signalling
    and potentially for heritable epigenetics.

    Histones have been studied for many decades as abundant proteins which were readily
    isolated due to their highly basic chemical character. Successive improvements in
    fractionation revealed 5 main types under a nomenclature as histone H1, H2A, H2B, H3
    and H4 \citep{nomenclature}. An additional histone H5 related to H1 is found in
    nucleus--containing avian erythrocytes and has been of research interest due to its
    elevated binding stability \citep{HFive-review}.
    Since the ratio of arginine/lysine is \LinkerArgLysRatio{}
    for H1 these became known as ``lysine-rich'', whereas the remaining 4 histones with
    ratios \CoreArgLysRatio{} are ``arginine-rich''.

    The demonstration of the nucleosome as the fundamental repeating unit of chromatin revealed
    that H2A, H2B, H3 and H4 associate as an octamer of two copies each within the
    nucleosome core particle, so they are referred to as ``core histones''. H1 and H5
    associate with the linker DNA between nucleosome core particles, and are referred to
    as ``linker histones''.

    Extracting histones from cells and separating them by polyacrylamide gel electrophoresis
    (PAGE) using the strongly anionic detergent sodium dodecyl sulphate (SDS) and neutral
    buffers gives single bands for each histone type. However, PAGE with acetic acid in gels
    containing the non-ionic detergent Triton X--100 and uncharged urea as denaturants
    (TAU or AUT) allows the separation of histones into multiple species due to both
    post-translational modifications and variations in sequence within types \citep{PAGEND}.

    There are two recognised classes of histone protein sequence variation as summarised
    in \tref{tab:typical-histone-differences}. ``Canonical'' histones contribute to the
    bulk structure and primary function of the nucleosome. Because its primary role is to
    package the genome, they are mainly expressed during S~phase for DNA replication,
    and so are referred as ``replication dependent''. In contrast, ``variant'' histones
    have lower aggregate abundance and show less identity in alignments. They play
    functionally distinctive roles, and most are significantly expressed outside S~phase
    of the cell cycle. Because of these, they are referred as ``replacement histones'',
    and ``replication independent''.

    \begin{table*}
      \caption{General properties of canonical and variant histone proteins.}
      \label{tab:typical-histone-differences}
      \centering
      \begin{tabular}{l l l}
        \toprule
        \null                     & Canonical             & Variants \\
        \midrule
        Expression timing         & Replication dependent & Replication independent \\
        Sequence identity         & High                  & Low \\
        Functional relationships  & Isoforms              & Distinct specialised functions \\
        Transcript stabilisation  & Stem--loop in 3' UTR  & poly--A tail \\
        Genome distribution       & Clusters              & Scattered \\
        \bottomrule
      \end{tabular}
    \end{table*}

    However, the term ``variant histones'' masks another level of variance among the canonical
    histones. Each of the variants is encoded by a single gene, with its own sequence and
    name. Examples of histone variants are H2A.Z, H2Abbd, and H3.3 \citep{variants-review}.
    On the other hand, canonicals are encoded by multiple genes with little varying sequence,
    and because their resulting protein isoforms differ at only a few sites, they are often
    referred to as ``wild type'' sequences. This has led to their portrayal as single histone
    types, with an implicit suggestion that they can be represented by a single sequence.
    When their multiple isoforms are not treated as a single protein, they are divided
    into``sub-types'' such as H2A.1 and H2A.2 but this division has no functional or sequence
    interpretation. To increase the confusion, the sub-types are also sometimes incorrectly
    referred to as ``variants''.

    The genes encoding canonical histone isoforms are typically found in clusters in most eukaryotic
    genomes, possibly for regulatory efficiency and to facilitate their sequence conservation. In
    contrast, most variant histones are encoded as single genes outside of the histone clusters.
    This distinction led to yet another distinct name, allelic and non-allelic variants for
    canonical isoforms and variant histones, respectively.

    The two classes of histones are also distinguished by their regulation in humans and other
    metazoans: like other genes transcribed by RNA polymerase II,  histone variants transcripts
    are polyadenylated on their 3' termini. Canonical histones are the only known transcripts
    that are not, terminating instead in a highly conserved stem-loop that is the basis for
    regulation of turnover. In general, canonical histone transcripts are not thought to be
    polyadenylated except in \textit{S. cerevisiae}.

    For the sake of brevity, and since canonical histones are the main subject of this
    publication, these are referred to as histones unless otherwise noted.

    Despite the importance of histones to the organisation of chromatin, and extensive interest
    in their role in epigenetics and regulation, the curation and classification of human histone
    gene and protein sequences has not been revisited for a decade \citep{Marzluff02}. In this time
    the draft human genome has been finalised and rich annotations have been propagated through
    gene and protein sequence databases. We have also scanned the popular RefSeq
    database to curate and report a number of oversights in untranslated regions, transcripts and
    classification of pseudo genes. A list of all the changes since \cite{Marzluff02} is 
    displayed on \tref{tab:difference-from-Marzluff02}.

    \begin{table*}
      \caption{Changes between the current status and the last published reference \citep{Marzluff02}.}
      \label{tab:difference-from-Marzluff02}
      \centering
      \input{results/table-reference_comparison}
    \end{table*}

    In this paper we provide a comprehensive overview of the nomenclature and distribution
    of canonical and variant histone genes. Furthermore, since curation and annotation are
    dynamic processes, we have implemented an accompanying resource that automatically generates
    the tables and figures presented in the submitted manuscript based
    on the most current data from the NCBI RefSeq database.

    \todo[inline]{We could shadow all the values and tables whose values are generated
                  automatically, so that readers can really an idea of how far this goes}

  %%%%%%%%%%%%%%%%
  %%
  %% Histone genes
  %%
  %%%%%%%%%%%%%%%%

  \section{Histone genes}
    %% section flow: we start describing the organization very broadly. Then we actually get a
    %% clusters subsection, comparing between other organisms, and the distribution of genes
    %% in each one. Only after this is known, can we explain gene nomenclature.

    The genes encoding canonical histone isoforms are typically found in clusters in most
    eukaryotic genomes, possibly for regulatory and expression functions and to facilitate
    sequence conservation. Indeed, histone transcripts require specialized machinery for
    their processing, which aggregate at the histone cluster locations during S~phase as
    histone locus bodies (HLB) \citep{human-first-HLB-report, human-HLB-report-GhuleEtAl2008}.
    These organelles are related to Cajal bodies, from whom they used to be indistinct, and
    whose division is still disputed \citep{CB-HLB-review}.

    \subsection{Histone gene clusters}
      In humans, there are 4 histone gene clusters, named histone cluster 1 to 4 (HIST1 to HIST4)
      by increasing order of their length. The major and secondary histone gene clusters,
      HIST1 and HIST2 on chromosomes~\HISTOneLocus{} and \HISTTwoLocus{} respectively,
      contain contiguous high density arrays of histone genes. On the other hand, the minor
      cluster HIST3 on chromosome~\HISTThreeLocus{} has only a handful of them, while HIST4
      on \HISTFourLocus{} can barely be considered a cluster with its single gene
      (\tref{tab:histone-gene-count}).

      \begin{table*}
        %% FIXME maybe we should generate the table automatically, instead of
        %%        generating all this variables. Or we could do the operations
        %%        in LaTeX with fp package:
        %%        \FPeval{result}{clip(\var1{}+\var2{})} \result
        \caption{Count of expected coding and pseudo histone genes.}
        \label{tab:histone-gene-count}
        \centering
        \begin{tabular}{l r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$}}
          \toprule
          \null   & \multicolumn{2}{c}{H2A}  & \multicolumn{2}{c}{H2B}
                  & \multicolumn{2}{c}{H3}   & \multicolumn{2}{c}{H4}
                  & \multicolumn{2}{c}{Total} \\
          \midrule
          HIST1   & \HISTOneHTwoACoding{}     & \HISTOneHTwoAPseudo{}
                  & \HISTOneHTwoBCoding{}     & \HISTOneHTwoBPseudo{}
                  & \HISTOneHThreeCoding{}    & \HISTOneHThreePseudo{}
                  & \HISTOneHFourCoding{}     & \HISTOneHFourPseudo{}
                  & \CodingGenesInHISTOne{}   & \PseudoGenesInHISTOne{} \\
          HIST2   & \HISTTwoHTwoACoding{}     & \HISTTwoHTwoAPseudo{}
                  & \HISTTwoHTwoBCoding{}     & \HISTTwoHTwoBPseudo{}
                  & \HISTTwoHThreeCoding{}    & \HISTTwoHThreePseudo{}
                  & \HISTTwoHFourCoding{}     & \HISTTwoHFourPseudo{}
                  & \CodingGenesInHISTTwo{}   & \PseudoGenesInHISTTwo{} \\
          HIST3   & \HISTThreeHTwoACoding{}   & \HISTThreeHTwoAPseudo{}
                  & \HISTThreeHTwoBCoding{}   & \HISTThreeHTwoBPseudo{}
                  & \HISTThreeHThreeCoding{}  & \HISTThreeHThreePseudo{}
                  & \HISTThreeHFourCoding{}   & \HISTThreeHFourPseudo{}
                  & \CodingGenesInHISTThree{} & \PseudoGenesInHISTThree{} \\
          HIST4   & \HISTFourHTwoACoding{}    & \HISTFourHTwoAPseudo{}
                  & \HISTFourHTwoBCoding{}    & \HISTFourHTwoBPseudo{}
                  & \HISTFourHThreeCoding{}   & \HISTFourHThreePseudo{}
                  & \HISTFourHFourCoding{}    & \HISTFourHFourPseudo{}
                  & \CodingGenesInHISTFour{}  & \PseudoGenesInHISTFour{} \\
          \addlinespace
          Total   & \TotalCodingHTwoA{}       & \TotalPseudoHTwoA{}
                  & \TotalCodingHTwoB{}       & \TotalPseudoHTwoB{}
                  & \TotalCodingHThree{}      & \TotalPseudoHThree{}
                  & \TotalCodingHFour{}       & \TotalPseudoHFour{} \\
          \bottomrule
        \end{tabular}
      \end{table*}

      HIST1 encodes \CodingGenesInHISTOne{}~functional genes over a span of \HISTOneSpan{}
      and is the second most gene dense region on the megabase scale of the the
      genome \citep{MHC-III-analysis}. It is located at the distal end of the major
      histocompatibility complex (MHC) in the extended class~I region at chromosome 6p21.31
      \citep{MHC-I-transcript, MHC-complete-sequencing-1999}. Whether the proximity to this
      highly polymorphic region has any effect on the recombination of HIST1 genes has not
      been studied yet. However, it has been suggested that proximity to HIST1 suppresses
      recombinations on HLA-A, a gene from the MHC Class~I region \citep{MHC-repressed-by-HIST}.

      All H1 genes are in HIST1.

      \todo[inline]{need to access ``Human histone genes are interspersed with members
                    of the Alu family and with other transcribed sequences'' (1982) by
                    Sierra, others, and Stein}

      The location 1q21.1 near HIST2 has been associated with deletion
      \citep{HISTTwo-prone-deletion-discovery}, and since then has been reported to be prone
      to recurrent rearrangement \citep{HISTTwo-prone-deletion-focus}. HIST2 is actually
      2 sub-clusters. Interestingly, seems that that the one closer to 1q21.1 is the one
      full of pseudogenes.

      Aside some sub-clustering \fref{fig:cluster-organization}), it appears that
      there is no apparent organization
      of the histones genes through the histone clusters, of vertebrates and
      lower eukaryotes. The clusters of invertebrates however, display tens to hundreds
      of tandem repeats of units, each with one copy of the five histones. This is the
      case of sea urchin and drosophila melanogaster. The dispersive organization
      in vertebrates suggests that histones expression is regulated by close elements
      \citep{close-regulators}.

      \begin{figure*}
        \centering
        \missingfigure{diagram of histones organization on clusters}
        \caption{Histone gene distribution on the histone clusters}
        \label{fig:cluster-organization}
      \end{figure*}

    \subsection{Histone gene nomenclature}
      Histone genes have a specific nomenclature that encodes the cluster number and its
      position relative to the other histones \citep{Marzluff02}. An histone gene
      symbol, HIST1H3B for example, is divided into 3 parts: the cluster (HIST1),
      the histone type (H3), and its position (B). The position letter is interpreted as
      its position relative to the other histones of the same type, in the same cluster.
      Since the start of the cluster is defined as the side closer to the centromere, the
      example HIST1H3B translates to the second H3 of the histone cluster, the
      second closest to the centromere (\fref{fig:nomenclature}).

      \begin{figure*}
        \centering
        \includegraphics[width=\textwidth]{nomenclature-schematic.pdf}
        %% FIXME need to make this subcaptions
        \caption{Histone gene nomenclature. a) canonical histone gene encode their position
                 on the genome on their symbol. b) variant histone symbols are disperse through
                 the genome and cannot be placed in relation to the others so their member
                 letters derive from historical reasons.}
        \label{fig:nomenclature}
      \end{figure*}

      Two exceptions were made to this rule: 1) A missing position letter on the symbol indicates
      no other histones of the same type on the cluster. This is the case of HIST3H2A, the
      single H2A gene in HIST3. 2) There is no human HIST1H2AF. It exists an Hist1h2af mouse
      histone though, and this exception is likely to have been a compromise to accommodate
      matching the human symbols to the mouse orthologs, and ease the cross of data between the
      two model organisms.

      This was the original nomenclature as approved by the HUGO Gene Nomenclature Committee (HGNC)
      when the human genome was being finalized. However, new histones have since been
      reported (\tref{tab:difference-from-Marzluff02}) which forced new exceptions on the
      nomenclature to keep its apparent organization. For example, HIST2H2AA was renamed
      HIST2H2AA3 and a new HIST2H2AA4 has been added. Formerly, there was no distinction
      between the symbols of pseudo and coding genes, but new pseudo genes are suffixed with
      PS\# by order of discovery. But the ``old'' pseudo genes retain their original symbols
      so the lack of suffix does not indicate a coding protein.

      The symbols for variant histones are much different and simpler. Due to their dispersion
      through the genome, no attempt was made to encode relative information on their name. In
      a similar fashion to H2AFX, symbols of histone variants are reduced to their histone type,
      followed by the letter F (standing for Family), and a set of characters that identified
      the variant, usually with historical or functional interpretation. For example, H2AFX and
      H2AFZ are named accordingly to their migration order in acetic acid-urea gels
      \citep{HTwoA-first-variants}.
      The only exception to this is the H3 variant CENP--A, since it is also the component A
      of the CENtromeric Protein, initially reported by \cite{CENPA-first-report} before being
      identified as an histone protein \citep{CENPA-copurifies-histones, CENPA-sequence-analysis}.

    \subsection{Gene promoters}
      \todo[inline]{We already have the 500 bp upstream of the TSS, check how the seqlogo
                    looks like. And then fix the annotation of the genes to include these
                    promoter elements}

      Because histone genes have been so well conserved throughout the evolution, they make
      excellent models to study computational methods, even for promoters
      \citep{histone-promoter-modeling}.

      HIST1 is actually split into 2 parts, and different promoters have been reported for each
      part \citep{HISTOne-different-transcripts}.

      Automatic analysis of the genome after training with histone genes, identifies
      several other genes that are involved in the cell replication with points
      towards a common regulation method. They still lack the stem-loop though.

    \subsection{Introns}
      Histone genes have no introns on their genes, probably to circumvent the
      requirement for primary transcript processing when histones must be rapidly
      produced at S~phase. Some variant histones have introns.

  \section{The transcripts}
    %% we must say mRNA instead of genes because: 1) we can't defined gene (Pearson.
    %% Genetics: what is a gene?. Nature (2006) vol. 441 http://dx.doi.org/10.1038/441398a )
    %% and 2) it's only for genes transcribed by pol II (the onf for mRNAs)
    Unlike all other eukaryotic mRNAs, histone transcripts are not polyadenylated at
    their 3' end. This process consists in the cleaving of pre-mRNA, and addition of a
    poly(A) tail which promotes the mRNA stability and is involved in its transport
    into the citoplasm \citep{mRNA-end-processing}. The alternative method used by
    histones is the stem-loop, a specialized mRNA secondary.

    \subsection{Stem-loop}
      The stem-loop is encoded shortly after the stop codon, between xxx and yyy base
      pairs, much earlier than the typical polyadenylation signal. The stem-loop sequence
      has been extremely well conserved throughout evolution but it appears that the
      sequence itself is not so important for its function as its actual structure
      \citep{stem-loop-structure}.

      \missingfigure{consensus of human stem-loop.}

      The stem-loop is followed by a purine-rich Histone Downstream Element (HDE)
      downstream of the stop codon. The stem-loop interacts with the Stem-Loop
      Binding Protein (SLBP) to stabilise the mRNA in S~phase \citep{SLBP-regulation}
      while the HDE interacts with U7 snRNA to direct efficient 3 end processing
      \citep{HDE-sequence}.


      This specific machinery for mRNA processing is a large part of increase on
      transcription during the S~phase since their expression is also regulated
      by the cell-cycle \addref and allow for the histone mRNA stability \addref.

      Still, we found some canonical histones with annotated poly(A) tails. These annotations
      are likely to have been done automatically. However, we were also able to identify
      some of them on EST libraries which would suggest that they do exist. Since
      this libraries are performed using oligo-dTs, the histone genes are highly
      expressed, and there's only a few occurrences of them, points that this is
      the exception rather than the rule.

      Variant histones on the other side all have a poly(A) tail with notable
      pseudo-exception of H2AFX which has both a stem-loop and a poly(A) tail,
      and transcribes 2 different mRNAs \citep{HTwoAX-transcripts}.

      In general, metazoan canonical histones
      are not thought to be polyadenylated, although \textit{S.\ cerevisiae} histone transcript stability
      appears to be regulated by polyadenylation instead of RNA structures.

    \subsection{mRNA expression levels}
      %% levels of mRNA of each histone gene in 2-3 different human cell lines (at least one primary cell line)
      Expression of the RD histone genes is highly regulated by the cell cycle (hence the RD name). Probably due
      to the cell high demand for histone proteins during DNA replication, their expression levels rise dramatically
      during this phase of the cell cycle. During the rest of the cell cycle, histone genes are still transcribed
      but with much lower levels.

      However, there is only a 5-fold increase in their transcription rate at S~phase, compared to the other phases
      of cell cycle so regulation acts strongly at the post-transcriptional level\addref.

      To note however, that each gene seems to have different levels of increased transcription, even between genes in the
      same cluster.

      RI histone genes have expression dependent on their needs (testis specific, DNA damage, etc).

      mention chicken histone genes, we can remove half of the set, and the other rise in expression to compensate.


  \section{The proteins}
    While some of the histone genes encode the same protein, this is not always the
    case. On the case of H3 and H4 the number of unique proteins are low, \TotalHThree{}
    and \TotalHFour{} respectively. For H2A and H2B this number is much higher, \TotalHTwoA{}
    and \TotalHTwoB{}.

    The numbering of these proteins amino acids also has a catch. The first methionine
    is cleaved after expression and as such, it is never counter when numbering them.

    Most of the PTM are in the histone tails but a few are within the histone fold (H3 T45 for example)

    \subsection{Histone Fold Domain}
      \todo[inline]{Is histone protein structure too much off topic?}

      All histone proteins have a common motif named the Histone Fold Domain (HFD).
      It is composed of a long $\alpha$ helix and two loops on its side that fold
      back two other shorter $\alpha$ helices. This structure allows the histones
      to dimerize with their pair (H3--H4 and H2A--H2B) and form the histone
      octamer. The central $\alpha$ helix of each dimer pair fits in the gap between
      the $\alpha$1 and $\alpha$2 helices crossing the pair $\alpha$2 helix. In
      addition to these, H3 and H2B display a short $\alpha$N and $\alpha$C helix
      on their terminals.

      The HFD is common to all histone and shared with other
      proteins such as the CENP--TWSC and NF--Y complexes. All of these are nuclear
      proteins that interact with DNA. In the case of NF--Y, these even replace
      the H2A--H2b histones during active conditions.

      Whether these proteins actually ever happen to duplicate the nucleosome functions...


    \subsection{H2A}
      \begin{TableAndFigure*}
        \captionof{table}{histone H2A protein consensus}
        \label{tab:H2A-consensus}
        \input{results/table-H2A-align}

        \includegraphics[width=\textwidth]{seqlogo_H2A.pdf}
        \captionof{figure}{WebLogo of all H2A sequences. The bottom line are the amino acids
                           of H2AFJ whose sequence is different from any of the H2A proteins}
        \label{fig:H2A-weblogo}
      \end{TableAndFigure*}

      \todo[inline]{comment on H2AFJ}

    \subsection{H2B}
      \begin{TableAndFigure*}
        \captionof{table}{histone H2B protein consensus}
        \label{tab:H2B-consensus}
        \input{results/table-H2B-align}

        \includegraphics[width=\textwidth]{seqlogo_H2B.pdf}
        \captionof{figure}{WebLogo of all H2B sequences.}
        \label{fig:H2B-weblogo}
      \end{TableAndFigure*}

    \subsection{H3}
      \begin{TableAndFigure*}
        \captionof{table}{histone H3 protein consensus}
        \label{tab:H3-consensus}
        \input{results/table-H3-align}

        \includegraphics[width=\textwidth]{seqlogo_H3.pdf}
        \captionof{figure}{WebLogo of all H3 sequences.}
        \label{fig:H3-weblogo}
      \end{TableAndFigure*}

    \subsection{H4}
      %% these should all be the same but they are not. And there's an H4 in HIST4
      %% which is also present in mouse, which means has been conserved  in evolution
      With the exception of HIST1H4G, all H4 genes seem to encode the same protein. Even HIST4H4 which is the
      only gene on that ``cluster''.
      \begin{TableAndFigure*}
        \captionof{table}{histone H4 protein consensus}
        \label{tab:H4-consensus}
        \input{results/table-H4-align}

        \includegraphics[width=\textwidth]{seqlogo_H4.pdf}
        \captionof{figure}{WebLogo of all H4 sequences.}
        \label{fig:H4-weblogo}
      \end{TableAndFigure*}


%  \section{SNP's}
%    \todo[inline]{writing is dependent on what is found}

  \section{Canonical isoforms vs.~variants}
    %% Characteristics of a variant are the fact they seem to have special functions. Emphasis
    %% that some canonical may have special functions which no one bothered to check
    %% variants with extra special nomenclature are outside the histone clusters and have poly-A tails (sometimes)
    %% Complain about the lack of information on the different isoforms
    \todo[inline]{Rules and exceptions to definition of canonical and variants. Also point
                that the word canonical suggest antecedence which is not necessarily true (ref to our
                paper and Henikoff's)}

    Variants are the most likely with a standard gene, are considered independent from each other,
    spread around the genome and its expression is mostly independent of cell cycle \todo{explain
    why mostly? Do it next paragraph when complaining the naming}. When people say H2AX that's
    a variant. Note, H2A.2 and H3.2 are not considered variants.

    Both this separation and their naming is very misleading \todo{say that the fact that no one cares is proof?}
    for several reasons.
    \begin{itemize}
      \item The word canonical suggests precedence which is not necessarily true. It has been
            suggested that at least in the case of H2A, the `canonical' gene may have evolved
            from one of the `variant' genes\addref.
      \item Not all `variant' are replication-independent. H2AX gene for example expression
            increases during S--phase together with other `canonical' histones and does show
            a transcript with stem-loop.
      \item The `variant' vs `canonical' separation suggests that there's no variance inside
            the `canonical' which is false. There's several unique protein sequences for each
            `canonical' protein. And even the sometime named H2A.2 and H3.2 variants are actually
            `canonical' histone proteins.
    \end{itemize}

  \section{Sequence extractor}
    To obtain the sequences for the alignment, curation, and IDs for the tables, a perl program using the
    BioPerl modules was written. This program performs a search for genes on entrez, retrieves
    their sequences from the reference assembly, mRNAs and proteins. The sequences are saved in separate files whose
    names are either the gene/transcript/protein name or id.

    The program (bp\_genbank\_ref\_extractor) was released under GPLv3 and is distributed with
    the Bio-EUtilities distribution (part of the BioPerl project) since version 1.73.
    \todo{actually, it's in the development version, next version might be 1.73 or 2.0 or whatever}

    \todo[inline]{the name bp\_genbank\_ref\_extractor is quite bad. We are still on time to change
                  it for something nicer before the release}

    and when appropriate we contributed to the upstream projects.

%    genes may not change a lot but this a good framework to still study the promotor


    \todo[inline]{some of our proposed annotations and gene status changes were refused}

  \section{Reproducible research}
  \label{sec:reproducible}
    While writing this document, we attempted to follow the principles of reproducible research
    \citep{reproducible-research-bioinformatics, reproducible-research-law}.
    We are also aware of the continuous development of databases and genetic knowledge. As such,
    the source code to replicate the tables and figures shown in this document are freely accessible
    online at \url{https://github.com/af-lab/histone-catalog}. We hope that in the future, it can
    be used to create up to date information
    that reflects the current knowledge on the human histone genes.

    The choice of RefSeq through Entrez over HAVANA through Ensembl was mainly based in one reason --- the ease
    to automate the gene search and retrieve their IDs. Entrez has made available E-utilities with a SOAP interface
    that is already part of the bioperl project. This allowed us to retrieve the search results very easily.
    However, the Ensembl search is done by a Lucene server which is not publicly accessible. Setting up our own
    Lucene instance to provide SOAP services from it was not straightforward for us. The Ensembl perl API is
    also not part of the bioperl project, needing to be installed separately, and is dependent on a very
    out of date version of Bioperl, further complicating its installation. All this would make reproducibility
    of our data a much more difficult process by other researchers.

%    We reason that the use of mature software development tools, could be used in other quantitative biology.

  \section{Acknowledgements}
    We are grateful to all the nice people at \#perl and \#bioperl channels on
    \url{irc.freenode.net}, especially Altreus, mst, LeoNerd, and pyrimidine for
    their help with the most technical programming details.

  \bibliographystyle{plainnat}
  \bibliography{references}

  %% this is a very long table (ctabular) that does not play nice with two columns
  %% so we place it at the end. See why it was a problem at
  %% http://tex.stackexchange.com/questions/101791/float-a-wide-ctabular-in-two-columns
  %% We could split the table in two parts in which case we could use begin{table*}
  \onecolumn
  \captionof{table}{Human canonical histone genes, transcripts and protein}
  \label{tab:histone-catalogue}
  \input{results/table-histone_catalogue}

\end{document}
