\documentclass[10pt,a4paper,onecolumn,article]{memoir}
\counterwithout{section}{chapter}
\DoubleSpacing

\usepackage[T1]{fontenc}
\usepackage{palatino}
\usepackage{graphicx}
  \graphicspath{{./figs/}}

%% remove colorlinks when preparing for print
\usepackage[final,hyperindex,hyperfootnotes,bookmarksnumbered,colorlinks]{hyperref}

\usepackage{eqparbox}
  %% keep size of first column of sequence align table on memory to calculate
  %% the size limit for the second column
  \newsavebox{\firstentrybox}
  \newcolumntype{F}{%
    >{\begin{lrbox}{\firstentrybox}}%
      l%
    <{\end{lrbox}%
    \eqmakebox[firstentry][l]{\unhcopy\firstentrybox}}}

\usepackage{fp}
\usepackage{fmtcount}
\usepackage{seqsplit}
\usepackage{multirow}
\usepackage{url}
\usepackage[textsize=footnotesize]{todonotes}
  %% new command for box about missing references
  \newcommand{\addref}[1][]{\todo[color=red!40,size=\tiny]{Add reference: #1}}

\usepackage{natbib}

%% to keep a pair of related table and figures, we set a new float and use it
%%conjunction with the captionof command from capt-of
\usepackage{capt-of}
\newfloat{TableAndFigure}{X}{capname}

\input{results/variables-align_results}
\input{results/variables-cluster_stats}
\input{results/variables-protein_stats}
\input{data/variables-Marzluff-stats}

\author{David Miguel Susano Pinto and Andrew Flaus}
\title{The canonical core histone catalogue}

\begin{document}
  %% Text flow: first introduction on what are histones and nucleosomes. Some history
  %% on them, and explanation on the different types. Then we list their organization,
  %% following the same pattern of gene expression: first describe genes and their
  %% organization, then their transcripts, and finally the proteins. One everything is
  %% presented, discuss on how the current division of histones is inappropriate and
  %% leads to misconceptions about histone variance. Finalize with comment on reproducible
  %% research and the code availability to rebuild all tables and analysis of this paper,
  %% and how we have used it for other organisms.

  \maketitle

  \begin{abstract}
    The core histone proteins are encoded by a large family of genes distributed in clusters across the
    genome. It is often overlooked that the canonical histone genes display variation within
    themselves, encoding \HTwoAUniqueProteins{} H2A, \HTwoBUniqueProteins{} H2B,
    \HThreeUniqueProteins{} H3, and \HFourUniqueProteins{} H4
    unique protein sequences. This canonical histone isoforms diversity exists
    in addition to the more divergent histone variants that perform specific chromatin roles.

    \todo{say something about nature of canonical variation}
    The most divergent isoforms differ in protein sequence for H2A by ?\%, H2B by ?\%, H3 vy ?\% and H4
    by ?\%. 

    Annotation of the human genome is in flux and is constantly being updated and enriched in reference
    databases such as RefSeq. To avoid the obsolescence of manual surveys but maintain the readability
    of scientific literature, we have developed a novel software framework that dynamically generates
    an up-to-date canonical histone catalogue based on human genome annotations and inserts the
    information into this paper format. We show how this can be readily adapted to histones of other
    organisms and applied to other gene families.
  \end{abstract}

  \include{sections/intro}
  \include{sections/genes}
  \include{sections/transcripts}
  \include{sections/proteins}

    %% TODO
%  \section{SNP's}

  \include{sections/discussion}
  \include{sections/computation}


  \section{Acknowledgements}
    We are grateful to all the nice people at \#perl and \#bioperl channels on
    \url{irc.freenode.net}, especially Altreus, mst, LeoNerd, and pyrimidine for
    their help with the most technical programming details.

  \bibliographystyle{plainnat}
  \bibliography{references}

  %% this is a very long table (ctabular) that does not play nice with two columns
  %% so we place it at the end. See why it was a problem at
  %% http://tex.stackexchange.com/questions/101791/float-a-wide-ctabular-in-two-columns
  %% We could split the table in two parts in which case we could use begin{table*}
  \onecolumn
  \captionof{table}{Human canonical histone genes, transcripts and protein}
  \label{tab:histone-catalogue}
  \input{results/table-histone_catalogue}

\end{document}
