\documentclass[10pt,a4paper,onecolumn,article]{memoir}
\counterwithout{section}{chapter}
\DoubleSpacing

\usepackage[T1]{fontenc}
\usepackage{palatino}
\usepackage{graphicx}
  \graphicspath{{./figs/}}

%% remove colorlinks when preparing for print
\usepackage[final,hyperindex,hyperfootnotes,bookmarksnumbered,colorlinks]{hyperref}

\usepackage{eqparbox}
  %% keep size of first column of sequence align table on memory to calculate
  %% the size limit for the second column
  \newsavebox{\firstentrybox}
  \newcolumntype{F}{%
    >{\begin{lrbox}{\firstentrybox}}%
      l%
    <{\end{lrbox}%
    \eqmakebox[firstentry][l]{\unhcopy\firstentrybox}}}

\usepackage{color}    % to color background of generated values values
\usepackage{seqsplit}
\usepackage{url}
\usepackage[textsize=footnotesize]{todonotes}
  %% new command for box about missing references
  \newcommand{\addref}[1][]{\todo[color=red!40,size=\tiny]{Add reference: #1}}

\usepackage{natbib}

%% to keep a pair of related table and figures, we set a new float and use it
%%conjunction with the captionof command from capt-of
\usepackage{capt-of}
\newfloat{TableAndFigure}{X}{capname}

\input{results/variables-align_results}
\input{results/variables-cluster_stats}
\input{results/variables-protein_stats}
\input{results/variables-utr}
\input{data/variables-Marzluff-stats}

%% We define a style here which will be used around all the values found
%% by the scripts. The idea is to place them in a grey box so that people
%% reading the paper can really get the idea that the values in the text
%% are automatically generated. The problem is that the variables are not
%% simple numbers or text, and we want to make operations with them here
%% in LaTeX. So we get an handy trick, and redefine FPeval.
%% See http://tex.stackexchange.com/questions/159155 for details
\newcommand{\ScriptValue}[1]{\setlength{\fboxsep}{2pt}\colorbox[gray]{0.8}{#1}}
\let\OldFPeval\FPeval
\def\FPeval{\renewcommand{\ScriptValue}[1]{##1}\OldFPeval}

\author{David Miguel Susano Pinto and Andrew Flaus}
\title{The canonical core histone catalogue}

\begin{document}
  %% Text flow: first introduction on what are histones and nucleosomes. Some history
  %% on them, and explanation on the different types. Then we list their organization,
  %% following the same pattern of gene expression: first describe genes and their
  %% organization, then their transcripts, and finally the proteins. One everything is
  %% presented, discuss on how the current division of histones is inappropriate and
  %% leads to misconceptions about histone variance. Finalize with comment on reproducible
  %% research and the code availability to rebuild all tables and analysis of this paper,
  %% and how we have used it for other organisms.

  \maketitle

  \begin{abstract}
	Core histone proteins are encoded by a large family of genes distributed
	in \todo{total histone clusters} clusters across the human genome.
	In total there are \TotalGenes{} canonical core histone genes, including
	\HTwoACodingGenes{} for H2A, \HTwoBCodingGenes{} for H2B,
	\HThreeCodingGenes{} for H3 and \HFourCodingGenes{} for H4, 
	along with at least \TotalPseudoGenes{} total pseudogenes.
	The histone genes display coding variation that gives rise to
	\HTwoAUniqueProteins{} H2A, \HTwoBUniqueProteins{} H2B, 
	\HThreeUniqueProteins{} H3 and \HFourUniqueProteins{} unique protein isoforms.
	This represents a surprising and seldom recognised variation
	of up to \todo{highest identity}\% amino acid identity in \todo{highest identity histone type}. 
	The isoform diversity exists in parallel to the well-known specialisation of variant histones 
	and supersedes commonly used designations such as H2A.1 and H3.1. 
	As a first step to clarifying whether specialised functions or expression profiles exist, 
	we have curated and catalogued the entire human canonical core histone gene and protein complement.
	This work has led to multiple corrections to human histone annotations, 
	reflecting the continual flux of the human genome as it is updated and enriched in reference databases. 
	To avoid the obsolescence of gene catalogues but maintain readability of scientific literature, we have developed 
	a novel software framework that dynamically generates the up-to-date canonical histone repertoire 
	based on the most current genome annotations and inserts the information into this manuscript format. 
	This approach can readily be adapted to histones of other organisms and to other gene families.
  \end{abstract}

  \input{sections/intro.tex}
  \input{sections/genes.tex}
  \input{sections/transcripts.tex}
  \input{sections/proteins.tex}

    %% TODO
%  \section{SNP's}

  \input{sections/discussion.tex}
  \input{sections/computation.tex}


  \section{Acknowledgements}
    We are grateful to all the nice people at \#perl and \#bioperl channels on
    \url{irc.freenode.net}, especially Altreus, mst, LeoNerd, and pyrimidine for
    their help with the most technical programming details.

  \bibliographystyle{plainnat}
  \bibliography{references}

  %% Supplementary tables and figures
  \onecolumn

  \newpage
  \captionof{table}{Differences between the current reference (as per RefSeq)
                    for human histone genes and the last published reference \citep{Marzluff02}.}
  \label{tab:difference-from-Marzluff02}
  \input{results/table-reference_comparison}

  %% this is a very long table (ctabular) that does not play nice with two columns
  %% so we place it at the end. See why it was a problem at
  %% http://tex.stackexchange.com/questions/101791/float-a-wide-ctabular-in-two-columns
  %% We could split the table in two parts in which case we could use begin{table*}
  \newpage
  \captionof{table}{Catalogue of all human canonical core histone genes, with
                    respective transcripts, protein, and IDs}
  \label{tab:histone-catalogue}
  \input{results/table-histone_catalogue}

  \newpage
  \captionof{table}{Catalogue of all human variant core histone genes, with
                    respective transcripts, protein, and IDs}
  \label{tab:variant-catalogue}
  \input{results/table-variant_catalogue}

  \newpage
  \input{results/histone_insanities.tex}

\end{document}
