\documentclass[a4paper,oneside,onecolumn,article,draft]{memoir}
\counterwithout{section}{chapter}

\input{preamble}
\input{preamble-manuscript}

\title{The Human Canonical Core Histone Catalogue}

\author[1]{David Miguel Susano Pinto}
\author[1]{Andrew Flaus}

\affil[1]{Centre for Chromosome Biology, School of Natural Sciences,
  National University of Ireland Galway, Galway, Ireland}

\begin{document}

  \maketitle

  \begin{abstract}
    Core histone proteins H2A, H2B, H3, and H4 are encoded
    by a large family of genes distributed across the human genome
    in \TotalCoreCodingGenes{} clusters
    with \TotalCoreCodingGenes{} coding genes comprising
    \HTwoACodingGenes{} for H2A,
    \HTwoBCodingGenes{} for H2B,
    \HThreeCodingGenes{} for H3,
    and \HFourCodingGenes{} for H4,
    along with at least \TotalCorePseudoGenes{} total pseudogenes.
    Canonical core histone genes display coding variation that gives rise to
    \HTwoAUniqueProteins{} H2A, \HTwoBUniqueProteins{} H2B,
    \HThreeUniqueProteins{} H3, and \HFourUniqueProteins{} H4
    unique protein isoforms.
    Although histone proteins are highly conserved overall,
    these isoforms represent a surprising variation
    exceeding commonly used subtype designations
    such as H2A.1 and H3.1 and they are not histone variants.
    This gene and protein complement is catalogued into manuscript format
    using software build automation tools and live genome annotations.
    Automatically generated values are shown with light grey background.
    This dynamic manuscript approach is inspired
    by the aims of reproducible research
    and can be readily adapted to other gene families.
  \end{abstract}

  \input{sections/intro.tex}
  \input{sections/genes.tex}
  \input{sections/transcripts.tex}
  \input{sections/proteins.tex}
  \input{sections/computation.tex}
  \input{sections/conclusion.tex}
  \input{sections/materialsmethods.tex}

  \section{Acknowledgements}
    We are grateful to the Perl community on the \url{irc.freenode.net}
    channels \#perl and \#bioperl,
    especially Altreus, mst, LeoNerd, and pyrimidine.
    Invaluable support was also provided by the \TeX{} stackexchange community,
    in particular David Carlisle.
    DMSP acknowledges the support of the Portuguese Foundation for
    Science and Technology (FCT).

  \bibliography{references}

  \newpage
  \appendix
  \input{sections/supplementary.tex}

\end{document}
