\section{Reproducible research}

  A considerable number of builds and updates of the human genome sequence have been made
  since the last major survey of human canonical histone genes in 2002 \citep{Marzluff02}.
  This has resulted in
  \AddedSinceReference{} canonical core histone genes added,
  \RemovedSinceReference{} removed,
  and \SequencesUpdatedSinceReference{} sequences
  being updated for the current RefSeq release (\tref{tab:difference-from-Marzluff02}).
  These changes reflect the ongoing nature and challenges of
  genome curation and annotation in reference databases \citep{BorkKoonin1998}.

  It is inevitable that sequences and annotations will continue to be revised
  based on continuous curation and improved experimental insights.
  This has recently been recognised as the pricess of ``biocuration'' \citep{biocuration2018}.
  For histones, changes in annotations in turn prompt 
  reevaluation of assumptions about their biological contributions,
  illustrated by the recognition that some genes
  annotated as canonical core histone isoforms are testes-specific variants \citep{Talbert2012},
  and by observations of cell type-specific expression \citep{YangGenomeBiol2011,Molden2015}.

  It is important for communities of researchers to contribute
  to ongoing formal curation of genomics resources
  by feeding back information to database maintainers \citep{SteinNRG2001},
  although many are unaware of this opportunity \citep{HollidaySPR2015}.
  In the course of this work we have suggested a significant number of proposals
  for improvements to RefSeq curators.
  These were identified automatically by scripted tests for consistency and uniformity
  between the database annotations and expected properties of histones.
  \tref{tab:curation-anomalies} lists the current apparent anomalies from these tests
  and is the basis for ongoing discussions with curators.

  Dynamic data can be presented in specialist online database resources
  such as the Histone Database \citep{HistoneDB2016}, 
  MS_HistoneDB \citep{ElkennaniGovin2017} and HIstome \citep{HIstome2012}.
  These database interfaces provide comprehensive access to data
  but have limited curation and are difficult to cite.
  They can also cease to be updated or become inaccessible.
  In contrast, manuscripts are the established method of scientific communication
  because they provide descriptive context and accessible formatting for readers,
  can be peer reviewed, and have well-established methods for citation.
  There are established mechanisms for permanent archival of published manuscripts.
  However, static catalogues in manuscripts
  such as earlier surveys of histone genes \citep{AlbigHumangen1997,Marzluff02}
  inevitably become supersed by improvements in source data and cannot be updated to remove errors.

  A self-updating manuscript bridges the features of dynamic database and static manuscript
  presentation styles by providing convenient access to the most current information.
  It encourages communities of researchers to directly feed into the formal curation process
  and enables rapid leveraging of the most current data for relatively stable gene families.

  Implementating a dynamic manuscript is also an example of ``reproducible research''
  \citep{reproducible-research-bioinformatics,reproducible-research-law}
  that can address the topical challenge of irreproducibility in biological data
  and interpretation \citep{ErrorProne2012,OpenPrograms2012}.

  This manuscript is generated directly from sequences and annotations
  in the core NCBI RefSeq resource \citep{OLearyRefseq2016}.
  The processing system for the manuscript does not cache intermediate information,
  so all changes contributed by the community of histone researchers
  and curated by professional database maintainers
  are directly and automatically reflected at each manuscript refresh.
  All scripts including instructions for automatic builds
  are transparently available in a public repository.
  Core processing is based on a BioPerl program contributed to the Bio-EUtilities distribution
  and publicly available since 2013.
  Dependencies on raw data sources, alignment algorithms, or display output libraries
  can also be upgraded since the processing is automatic and script-based.

  One potential challenge for researchers is the referencing of dynamic data within such a manuscript.
  This can be simply addressed by users citing the publication as a traditional static manuscript
  then stating the build date in Materials and Methods, as they would for a database.

  Although the dynamic data will remain current,
  major revisions in understanding of a gene family or accumulation of small insights
  can in time render the explanatory static manuscript text obsolete.
  In this case the manuscript can simply be refreshed and republished independently
  in the same way as a traditional manuscript would be.
  The underlying scripts generating the dynamic data do not need to be rewritten,
  although new functions can be added to reflect new insights.

  The core scripts underlying this manuscript have been written to facilitate generating
  equivalent catalogues for other organisms via simple build options.
  We have successfully trialled this for \textit{Mus musculus} (not shown)
  to demonstrate that an equivalent dynamic manuscript for other histone gene sets
  only requires drafting appropriate surrounding static explanatory text.
  The approach is also applicable to larger and more diverse gene families such as
  our previous cataloguing of the Snf2 family \citep{andrew-snf2-catalogue}.
