\section{Reproducible research}
\label{sec:reproducible}

  A considerable number of builds and updates of the human genome sequence have been made
  since the last major survey of human canonical histone genes in 2002 \citep{Marzluff02}
  resulting in
  \AddedSinceReference{} canonical histone genes added,
  \RemovedSinceReference{} removed,
  and \SequencesUpdatedSinceReference{}
  sequences updated (\tref{tab:difference-from-Marzluff02}).
  These changes reflect the ongoing nature and challenges of
  genome curation and annotation \citep{BorkKoonin1998}.

  The human canonical histone gene complement is relatively mature
  and is unlikely to undergo major change.
  Nevertheless, it is inevitable that sequences and annotations will be revised over time.
  This is illustrated by the recent recognition that
  some genes annotated as canonical isoforms are testes-specific variants \citep{Talbert2012},
  and by new observations of cell type-specific expression \citep{Molden2015}.

  It is important for communities of researchers to contribute
  to this curation and feed back information \citep{SteinNRG2001}, 
  although many are unaware of this opportunity \citep{HollidaySPR2015}.
  During this work we have submitted a number
  of proposals for incorrect, inconsistent, or missing annotations
  to RefSeq curators.  All of them were identified automatically
  by our scripted tests to match existing
  annotations to expected properties of histones. Outstanding
  anomalies are listed in \tref{tab:curation-anomalies}.

  Dynamic data is frequently presented in specialist online database resources
  but these can have restrictive interfaces and presentation styles, and are difficult to cite.
  In contrast, manuscripts are the established method of scientific communication
  because they provide comprehensibility, peer review, and established methods for citation.

  A self-updating manuscript bridges the dynamic database and static manuscript 
  presentation styles to provide convenient access to the most current information.
  It encourages communities of researchers to directly feed into the annotations
  and enables them to rapidly make use of data for any relatively stable gene family.

  One challenge for readers is to reference dynamic data within the manuscript. 
  This is simply addressed by citing the publication as a traditional static manuscript
  and stating the access date in Materials and Methods, as they would for a database.

  Implementating a dynamic manuscript is also an example of ``reproducible research'' 
  \citep{reproducible-research-bioinformatics,reproducible-research-law}
  that addresses the topical challenges of irreproducibility in biological data
  and interpretation \citep{ErrorProne2012,OpenPrograms2012}.
  The manuscript is generated directly from sequences and annotations
  in the core NCBI RefSeq resource \citep{PruittRefseq2014}.
  The processing system does not cache intermediate information, 
  so all changes contributed by the community of histone researchers 
  and curated by professional database maintainers
  is directly and automatically reflected in the next manuscript refresh.
  All scripts including instructions for automatic builds
  are transparently available in a public github repository.
  Core processing is based on a BioPerl script contributed to the Bio-EUtilities distribution
  and publicly available since 2013.
  Dependencies on raw data sources, alignment algorithms or display output libraries
  can also be upgraded since the processing is automatic and script-based.

  In the event of a major revision in the histone gene set such as removal of a cluster
  or eventual accumulation of insights rendering the explanatory text obsolete,
  the manuscript can be redrafted as a fresh publication
  without the requirement to repeat the underlying data generation scripts.
  New functions can also be added to scripts to reflect enhanced information
  such as variation from genome sequencing of multiple individuals
  while still retaining backwards compatibility.

  Furthermore, the core scripts are written to enable them to generate
  equivalent information for other organisms with a one line change.
  We have successfully trialled this for \textit{Mus musculus}
  and \textit{Gallus gallus} (not shown)
  so an equivalent dynamic manuscript on these canonical gene sets
  simply requires redrafting the surrounding explanatory text.
  The approach could also be applied to larger and more diverse gene families such as
  our previous cataloguing of the Snf2 family \citep{andrew-snf2-catalogue}.
