\section{Reproducible research}

  A considerable number of builds and updates of the human genome sequence have been made
  since the last major survey of human canonical histone genes in 2002 \citep{Marzluff02}
  resulting in
  \AddedSinceReference{} canonical core histone genes added,
  \RemovedSinceReference{} removed,
  and \SequencesUpdatedSinceReference{}
  sequences updated (\tref{tab:difference-from-Marzluff02}).
  These changes reflect the ongoing nature and challenges of
  genome curation and annotation \citep{BorkKoonin1998}.

  The human canonical core histone gene complement is relatively mature
  and is unlikely to undergo major change.
  Nevertheless, it is inevitable that sequences and annotations will be revised over time
  based on continuous curation and improved experimental insights.
  This in turn prompts reevaluation of assumptions about their properties,
  illustrated by the recognition that some genes
  annotated as canonical core histone isoforms are testes-specific variants \citep{Talbert2012},
  and by new observations of cell type-specific expression \citep{Molden2015}.

  It is important for communities of researchers to contribute
  to ongoing curation and to feed back information to database maintainers \citep{SteinNRG2001},
  although many are unaware of this opportunity \citep{HollidaySPR2015}.
  During this work we have submitted a number of proposals
  for incorrect, inconsistent, or missing annotations to RefSeq curators.
  These annotation issues were identified automatically
  by scripted tests for consistency and uniformity
  between the database annotations and expected properties of histones.
  \tref{tab:curation-anomalies} lists current apparent anomalies from these tests
  and is the basis for ongoing feedbackd to RefSeq.

  Dynamic data can be presented in specialist online database resources
  such as the Histone Database \citep{HistoneDB2016} and HIstome \citep{HIstome2012}.
  These database interfaces provide comprehensive access to data
  but have limited curation and are difficult to cite.
  In contrast, manuscripts are the established method of scientific communication
  because they provide descriptive context and accessible formatting for readers,
  can be peer reviewed, and have well-established methods for citation.
  However, static manuscripts such as earlier surveys of histone genes \citep{AlbigHumangen1997,Marzluff02}
  inevitably become supersed by improvements in source data and cannot be updated to remove errors.

  A self-updating manuscript bridges the features of dynamic database and static manuscript
  presentation styles by providing convenient access to the most current information.
  It encourages communities of researchers to directly feed into the annotation process
  and enables rapid leveraging of the most current data for relatively stable gene families.

  Implementating a dynamic manuscript is also an example of ``reproducible research''
  \citep{reproducible-research-bioinformatics,reproducible-research-law}
  that can address the topical challenge of irreproducibility in biological data
  and interpretation \citep{ErrorProne2012,OpenPrograms2012}.

  This manuscript is generated directly from sequences and annotations
  in the core NCBI RefSeq resource \citep{OLearyRefseq2016}.
  The processing system does not cache intermediate information,
  so all changes contributed by the community of histone researchers
  and curated by professional database maintainers
  are directly and automatically reflected in the next manuscript refresh.
  All scripts including instructions for automatic builds
  are transparently available in a public repository.
  Core processing is based on a BioPerl program contributed to the Bio-EUtilities distribution
  and publicly available since 2013.
  Dependencies on raw data sources, alignment algorithms, or display output libraries
  can also be upgraded since the processing is automatic and script-based.

  One potential challenge for researchers is the referencing of dynamic data within the manuscript.
  This can be simply addressed by users citing the publication as a traditional static manuscript
  then stating the access date in Materials and Methods, as they would for a database.

  Although the dynamic data will remain current,
  major revisions in understanding of a gene family or accumulation of small insights
  may in time render the explanatory static manuscript text obsolete.
  In this case the manuscript can simply be refreshed and republished in revised form
  in the same way as a traditional manuscript.
  The underlying scripts generating the dynamic data do not need to be rewritten,
  although new functions can be added to reflect new insights.

  The core scripts underlying this manuscript are written to facilitate generating
  equivalent catalogues for other organisms via build options.
  We have successfully trialled this for \textit{Mus musculus} (not shown).
  Therefore, an equivalent dynamic manuscript for such gene sets
  only requires redrafting the surrounding static explanatory text.
  The approach is also applicable to larger and more diverse gene families such as
  our previous cataloguing of the Snf2 family \citep{andrew-snf2-catalogue}.
