\section{Histone genes}

\subsection{Histone gene clustering}
	%% For simplicity, decided to list HIST4 explicitly for rather than allow it to come in by a variable
	The human canonical histone genes are located in clusters HIST1 to HIST4, 
	named in order of decreasing histone gene count (\tref{tab:histone-gene-count}).
	HIST1, is the major histone gene cluster at locus~\HISTOneLocus{}  
	with \CodingGenesInHISTOne{}~functional core histone genes plus all canonical H1 histones,
	representing \calc{\CodingGenesInHISTOne{}*100/\TotalCodingGenes{}}\% of all canonical core histone genes. 
	HIST2 at locus~\HISTTwoLocus{} contains \CodingGenesInHISTTwo{}~coding genes, 
	HIST3 at locus \HISTThreeLocus{} contains \CodingGenesInHISTThree{} coding genes, 
	and HIST4 at locus \HISTFourLocus{} contains only \CodingGenesInHISTFour{}~coding gene. 

	Although the definition of canonical core histone genes appears to be stable, 
	progress in understanding histone gene function may lead to changes in these analyses. 
	For example, although \textit{HIST1H2BA} is nominally a canonical histone 
	by nomenclature and cluster assignment
	it has received significant attention as testes-specific TH2/TH2B/hTSH2B \citep{Zalensky2002,LiAusio2005,Shinagawa2014}.
	This gene encodes the most divergent canonical H2B 
	and is located 300 kbp upstream of the rest of the HIST1 cluster 
	separated by a number of non-histone genes. 
	It is adjacent to \textit{HIST1H2AA} which encodes a H2A protein of similarly high variation, 
	and the syntenic ratorthologoues are divergently transcribed specifically in testes \citep{HuhChae1991}. 
	It is therefore possible that \textit{HIST1H2AA} and \textit{HIST1H2BA} 
	are undergoing sub-functionalisation and could be reclassified 
	as testes specific histone variants.

	A similar case could apply for the small HIST3 cluster 80 Mbp downstream of HIST2 
	which contains protein coding genes \textit{HIST3H2A}, \textit{HIST3H2BB} and \textit{HIST3H3}.
	\textit{HIST3H3} encodes testes-specific H3T/H3.4
	that has distinctive biochemical properties \citep{WittExpCellRes1996,KurumizakaCOSB2013}, 
	while \textit{HIST3H2A} and \textit{HIST3H2BB} are amongst the most divergent canonical histones of their types.

	HIST1 and HIST2 are both contiguous high density arrays of histone genes. 
	HIST1 spans \HISTOneSpan{} \todo{need space between number and unit}
	and is the second most gene dense region in the human genome at megabase scale \citep{MHC-III-analysis}.
	\textit{HFE} encoding the hemochromatosis protein \citep{AlbigDoenecke1998} 
	is the only non-histone protein coding gene located within the cluster
	excluding the space between HIST1 and \textit{HIST1H2AA}-\textit{HIST1H2BA}. 
	The implications of tight and exclusive histone gene clustering 
	is not well understood but may faciltate coordinate regulation \citep{Eirinlopez2009,close-regulators}.

	HIST1 is located towards the distal end of the major histocompatibility complex (MHC) 
	in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999}, 
	and it has been suggested that proximity to HIST1 
	suppresses recombination in the MHC Class~I region \citep{MHC-repressed-by-HIST}. 
	In contrast, it has been suggested that HIST2 is prone to 
	deletions and frequent rearrangements \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}
	although it is not thought that histone clustering contributes to gene conversion \citep{NeiRooney2005}.

\subsection{Histone gene sequences}
	Despite the high conservation of canonical histone proteins, 
	the coding regions their genes exhibit considerable variation (\fref{fig:histone-gene-variation}). 
	These differences are largely located in the third base position of codons (\fref{fig:histone-gene-variation})
	and reflect the very high ratio of synonymous to non-synonymous substitutions (\fref{fig:histone-gene-differences}).

	The mean number of synonymous codon differences between pairs of canonical isoform genes is 
	\todo{H2A coding gene synonymous diffs} for H2A, 
	\todo{H2B coding gene synonymous diffs} for H2B, 
	\todo{H3 coding gene synonymous diffs} for H3, 
	and \todo{H4 coding gene synonymous diffs} for H4.
	This is consistent with observations that 
	synonymous codon divergence exceeds non-synonymous variation 
	for histone genes across eukaryotes \citep{Piontkivska2002, Rooney2002}. 
	It supports a hypothesis that histone protein sequence conservation 
	results from birth and death evolution through strong selective pressure at the protein level \citep{NeiRooney2005}, 
	and not from sequence homogenisation based on clustering.

	Despite the level of synonymous substitution, 
	codon usage is strongly biased towards the most frequently used human codons for most amino acids (\fref{fig:histone-gene-codonusage}). 
	This suggests that histone translation continues to explore synonymous alternatives 
	despite an underlying dependence on the most abundant charged tRNAs.

\subsection{Canonical histone gene nomenclature}
	Canonical histone genes adhere to a Human Genome Organisation (HUGO) Gene Nomenclature Committee (HGNC) 
	endorsed system derived from the cluster number and position relative to other histones \citep{Marzluff02}. 
	This superseded an earlier arbitrary scheme with backslashes (e.g. H2b/b) 
	that preceded genome sequencing \citep{AlbigGenomics1997,AlbigHumangen1997}.

	The canonical histone gene symbols are divided into 3 parts: 
	HIST cluster, histone type, and identifier letter
	for the order relative to other histone genes of the same type in the cluster (\fref{fig:nomenclature}).
	For example, \textit{HIST1H2BD} is nominally the fourth H2B coding gene in the HIST1 cluster.
	HIST1 identifiers are ordered from telomere to centromere 
	whereas HIST2 and HIST3 are ordered from the centromere to telomere \citep{Marzluff02}.

\begin{figure*}
	\centering
	\includegraphics[width=\textwidth]{nomenclature-schematic-v2.pdf}
	\caption{Histone gene nomenclature. 
	Canonical histone gene names encode relative genomic order by cluster.
	Canonical pseudogenes named since 2002 include cluster, PS label and discovery order identifier,
	Most variant histone genes are identified with F and identifier letter.}
	\label{fig:nomenclature}
\end{figure*}

	Two exceptions were originally applied to these simple naming rules \citep{Marzluff02}. 
	Firstly, the positional identifier is omitted if there are no other histones of the same type in the cluster, 
	so \textit{HIST3H2A} is the sole H2A gene in HIST3. 
	Secondly, the human and mouse histone clusters are largely syntenic 
	so positional identifier letters for missing orthologues were skipped to maintain the equivalence of gene symbols.
	Consequently there is no human \textit{HIST1H2AF} although \textit{Hist1h2af} exists in mouse.

	Several new histone genes have been uncovered since the original naming (\tref{tab:difference-from-Marzluff02}) 
	and this required additional nomenclature exceptions.
	For example, new H2A encoding genes identified in the human and mouse HIST2 cluster preceding \textit{HIST2H2AA} 
	required renaming of \textit{HIST2H2AA} to \textit{HIST2H2AA3} 
	and the addition of a new human gene as \textit{HIST2H2AA4}. 
	There are no human orthologues of mouse \textit{Hist2h2aa1} and \textit{Hist2h2aa2}.

	Furthermore, no distinction was originally made between pseudogenes and functional coding genes,
	so \textit{HIST3H2BA} is a pseudogene whereas neighbouring \textit{HIST3H2BB} is the only H2B coding gene in HIST3.
	Newly uncovered pseudogenes are now suffixed with PS and a number in order of discovery,
	such as \textit{HIST1H2APS5} (\fref{fig:nomenclature}). 
	However, the pseudogenes named in the original classification retain their symbols without PS.
	This means that the absence of a PS suffix does not indicate a functional gene, 
	and that there is no positional information in the gene symbols of most pseudogenes.

\subsection{Histone variant genes and nomenclature}
	The set of annotated human histone variant genes 
	includes \todo{Number of variant genes} members 
	which are listed in \tref{tab:variant-catalogue} for completeness.
	Histone variant gene families comprise only one or a few copies 
	dispersed across the genome outside the canonical histone clusters.
	For example the three H3.3 variant encoding genes are located 
	on chromosomes 1, 12 and 17 far removed from other histone genes.

	Most variant gene symbols have a separate 3 part nomenclature 
	consisting of labels for histone type, F for family, 
	and an identifier letter based on discovery order (\fref{fig:nomenclature}).
	Increasing interest in histone variant function \citep{MazeAllis2014}
	coupled with a variety of usages and conflicts between species 
	has led to guidelines for improved consistency in histone variant nomenclature 
	\citep{Talbert2012} (\tref{tab:histone-variant-families}).
