\section{Histone genes}

  \subsection{Histone gene clusters}
	%% For simplicity, decided to list HIST4 explicitly for rather than allow it to come in by a variable
    The human genome contains histone gene clusters HIST1 to HIST4 
	named in order of decreasing histone gene count (\tref{tab:histone-gene-count}).
	The major histone gene cluster HIST1 at chromosome~\HISTOneLocus{}, 
	contains \todo{percent genes in HIST1}\% of all histone genes 
	including \CodingGenesInHISTOne{}~functional core histone genes plus all canonical H1 histones. 
	HIST2 at chromosome~\HISTTwoLocus{} contains \CodingGenesInHISTTwo{}~coding genes 
	and is also highly gene dense, spanning \HISTTwoSpan{} across \HISTTwoLocus{}. 
	HIST3 at \HISTThreeLocus{} and HIST4 at \HISTTFourLocus{} 
	contain only \CodingGenesInHISTThree{} and \CodingGenesInHISTFour{}~coding genes respectively. 
	Histone variants are dispersed across the genome outside the clusters, 
	so for example the three H3.3 encoding genes are located on chromosomes 1, 12 and 17 
	each removed from any other histone genes.

	HIST1 and HIST2 both contain contiguous high density arrays of histone genes. 
	HIST1 spans \HISTOneSpan{} \todo{need space between number and unit}
	and is the second most gene dense region in the genome at megabase scale \citep{MHC-III-analysis}.
	If \textit{HIST1H2AA} and \textit{HIST1H2BA} are excluded (see below), 
	the only non-histone protein encoded within HIST1 is the hemochromatosis gene \textit{HFE} 
	and there are no non-histone proteins at all within the HIST2 or HIST3 clusters. 

	HIST1 at \HISTOneLocus{} is located towards the distal end of the major histocompatibility complex (MHC) 
	in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999}. 
	It has been suggested that proximity to HIST1 can suppress recombination in the MHC Class~I region \citep{MHC-repressed-by-HIST}, 
	and it has also been reported that HIST2 is prone to deletions and frequent rearrangements \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}. 

	The implications of this tight exclusive clustering is not well understood, 
	although the dispersive organisation has been suggested to imply that expression is regulated by local elements \citep{close-regulators}. 

	%% Removed birth and death because it is not directly relevant to this data
	%% Removed Alu stuff because it has no biological implications
	%% Definitely need to keep these observation, maybe put in thesis discussion?

  \subsection{Histone gene nomenclature}
    Canonical histone gene names adhere to a Human Genome Organisation (HUGO) Gene Nomenclature Committee (HGNC) 
	endorsed system derived from the cluster number and position relative to other histones \citep{Marzluff02}.

	%% Should we try putting a url reference into the manuscript?
	%% http://www.genenames.org/genefamilies/histones
	The canonical histone gene symbol is divided into 3 parts: 
	HIST cluster, histone type, and positional identifier letter
	for the order relative to other histone genes of the same type in the cluster (\fref{fig:nomenclature}A).
	For example, \textit{HIST1H2BD} is the fourth H2B gene in the HIST1 cluster.
	The direction of the cluster order varies, so HIST1 is ordered from the centromeric side 
	whereas HIST2 and HIST3 are ordered from the telomeric side.
	%% Changed example to match figure!
	
    Two exceptions were originally applied to this simple gene naming rule \citep{Marzluff02}. 
	Firstly, the position letter is omitted if there are no other histones of the same type in the cluster, 
	so \textit{HIST3H2A} is the sole H2A gene in HIST3. 
	Secondly, the human and mouse histone clusters are largely syntenic 
	so positional identifier letters were skipped to maintain the equivalence of gene symbols 
	in genomes missing a particular expevcted orthologue.
	This means there is no human \textit{HIST1H2AF} although \textit{Hist1h2af} exists in mouse.

	Unfortunately several new histone genes have been uncovered since the original naming (\tref{tab:difference-from-Marzluff02}). 
	and this has forced additional nomenclature exceptions to maintain the original organisation.
	For example, \textit{HIST2H2AA} was renamed \textit{HIST2H2AA3} and a new \textit{HIST2H2AA4} was added
	because there are no human orthologues of mouse \textit{Hist2h2aa1} and \textit{Hist2h2aa2}.
	
	Furthermore, no distinction was originally made between pseudogenes and functional coding genes, 
	so \textit{HIST3H2AA} is a pseudogene whereas its neighbour \textit{HIST3H2AB} is a coding gene.
	Newly uncovered pseudogenes have since been suffixed with PS\# in order of discovery, such as \textit{HIST1H2APS2}. 
	The ``original'' pseudogenes retain their symbols, so the absence of a PS suffix does not indicate a functional gene.

	There remains the possibility of further change 
	through increased understanding of functional roles for specific canonical histone genes. 
	For example, \textit{HIST1H2BA} is now recognised as testes-specific TH2 \addref{Horokoshi ACTA Cryst 2014?} 
	and has the most divergent sequence of current canonical H2B proteins.
	It is adjacent from the divergently transcribed \textit{HIST1H2AA} gene 
	that is itself the most unique canonical H2A protein sequence, 
	although there is little evidence yet for a specific function. 
	The \textit{HIST1H2AA} and \textit{HIST1H2BA} gene pair 
	is located ~300 kbp upstream of the other HIST1 cluster members 
	separated by a number of unrelated non-histone genes. 
	It is conceivable that these two canonical histone genese
	could be removed from the HIST1 cluster list and reclassified as variant histones.
	
	In contrast with the human canonical core histone genes, 
	all recognised histone variant gene familes comprise only one or a few copies 
	dispersed across the genome outside clusters.
	These histone variant genes are identifed by a separate 3 part nomenclature 
	consisting of labels for histone type, F for family, 
	and an identifier letter based on discovery order (\fref{fig:nomenclature}B).
	Where two isoforms exist the copies can also have a numerical suffix, 
	although this is sometimes omitted for the first copy. 
	For example, \textit{H2AY} and \textit{H2AY2} encode the human macroH2As, 
	with \textit{H2AY} giving rise to alternative transcripts encoding macroH2A1.1 and macroH2A1.2 protein isoforms 
	whereas \textit{H2AY2} encodes macroH2A2.
	The variant histone gene family naming (\tref{tab:histone-variant-families}) also includes exceptions. 
	For example, the three gene copies encoding the H3.3 variant 
	take on a distinguishing letters \textit{H3F3A}, \textit{H3F3B} and \textit{H3F3C} instead of numbers 
	and the H3 variant gene CENPA  was named as the first identified centromeric protein \cite{CENPA-first-report} 
	before it was realised to be a histone protein \citep{CENPA-copurifies-histones, CENPA-sequence-analysis}. 
	Histone variant pseudogenes are suffixed with P to yield symbols such as \textit{H2AFXP} and \textit{H2AFZP5}.
	Histone variants have become of increasing functional interest \addref{Maze Allis NRG 2014}
	but their nomenclature have suffered imprecise usage and conflicts between species, 
	so guidelines have been proposed to improve consistency \addref{Talbert Epigen Chrom 2012}.

    \begin{figure*}
      \centering
      \includegraphics[width=\textwidth]{nomenclature-schematic.pdf}
      \caption{Histone gene nomenclature. 
	  a) Canonical histone gene names encode their relative genomic order by cluster. 
	  b) Variant histone symbols are identified with F then a family identifier letter
	  and numbered by discovery, except 1 is omitted for the founder member. 
	  H3.3 genes conversely use numbers and letters for family and member.
	  c) Canonical seudogenes identified since 2002 are named by cluster with PS identifier numbered by discovery. 
	  Previously canonical pseudogenes were named like functional histone genes.}
      \label{fig:nomenclature}
	  \todo{Parts A and B need labels. Name text need to be italics!}
	  \todo{Change part B to use H2AFY2 as a more complete example of variant nomenclature}
	  \todo{Include part C using HIST1H2APS2 as example of PS pseudogene nomenclature}
    \end{figure*}

%% Let's leave the promoters and introns because we don't have anything to say

  \begin{table*}
    \caption{HGNC recognised histone variant family stem names, commonly used protein names 
	and protein names for improved consistency proposed by Talbert et al (2012). 
	HIST1H2BA is nominally not a variant but is included for completeness.}
    \label{tab:histone-variant-families}
    \centering
    \begin{tabular}{l l l}
      \toprule
      \Family & Protein name & Talbert et al & Notes \\
      \midrule
	  H2AFB & H2A.Bbd & H2A.B & equivalent to H2A.L \\
	  H2AFJ & H2A.J & ? & ? \\
	  H2AFV & H2A.Z-2 & H2A.Z.1 & - \\
	  H2AFX & H2AX & H2A.X & DNA damage related \\
	  H2AFY & macroH2A/mH2A & macroH2A & - \\
	  H2AFZ & H2A.Z & H2A.Z.1 & - \\
	  H2BFM & ? & ? & - \\
	  H2BFS & - & - & pseudogene \\
	  H2BFWT & ? & H2B.W & testes specific \\
	  H2BFX & - & - & pseudogene \\
	  HIST1H2BA & TH2B/TSH2B & H2B.1 & testes specific \\
	  H3F3 & H3.3 & H3.3 & - \\
	  CENPA & CENP-A & ? & centromere-specific \\
	\bottomrule
    \end{tabular}
  \end{table*}

%% This is arguably the most important table. We should cite it as early as possible
    \begin{table*}
      \caption{
	  Count of human canonical histone coding and pseudogenes by histone cluster and type.
	  The right-hand value in each cell marked with $\psi$ is the number of pseudo genes.}
      \label{tab:histone-gene-count}
      \centering
      \begin{tabular}{l r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$}}
        \toprule
        \null   & \multicolumn{2}{c}{H2A}  & \multicolumn{2}{c}{H2B}
                & \multicolumn{2}{c}{H3}   & \multicolumn{2}{c}{H4}
                & \multicolumn{2}{c}{Total} \\
        \midrule
        HIST1   & \HTwoACodingInHISTOne{}     & \HTwoAPseudoInHISTOne{}
                & \HTwoBCodingInHISTOne{}     & \HTwoBPseudoInHISTOne{}
                & \HThreeCodingInHISTOne{}    & \HThreePseudoInHISTOne{}
                & \HFourCodingInHISTOne{}     & \HFourPseudoInHISTOne{}
                & \CodingGenesInHISTOne{}     & \PseudoGenesInHISTOne{} \\
        HIST2   & \HTwoACodingInHISTTwo{}     & \HTwoAPseudoInHISTTwo{}
                & \HTwoBCodingInHISTTwo{}     & \HTwoBPseudoInHISTTwo{}
                & \HThreeCodingInHISTTwo{}    & \HThreePseudoInHISTTwo{}
                & \HFourCodingInHISTTwo{}     & \HFourPseudoInHISTTwo{}
                & \CodingGenesInHISTTwo{}     & \PseudoGenesInHISTTwo{} \\
        HIST3   & \HTwoACodingInHISTThree{}   & \HTwoAPseudoInHISTThree{}
                & \HTwoBCodingInHISTThree{}   & \HTwoBPseudoInHISTThree{}
                & \HThreeCodingInHISTThree{}  & \HThreePseudoInHISTThree{}
                & \HFourCodingInHISTThree{}   & \HFourPseudoInHISTThree{}
                & \CodingGenesInHISTThree{}   & \PseudoGenesInHISTThree{} \\
        HIST4   & \HTwoACodingInHISTFour{}    & \HTwoAPseudoInHISTFour{}
                & \HTwoBCodingInHISTFour{}    & \HTwoBPseudoInHISTFour{}
                & \HThreeCodingInHISTFour{}   & \HThreePseudoInHISTFour{}
                & \HFourCodingInHISTFour{}    & \HFourPseudoInHISTFour{}
                & \CodingGenesInHISTFour{}  & \PseudoGenesInHISTFour{} \\
        \addlinespace
        Total   & \HTwoACodingGenes{}       & \HTwoAPseudoGenes{}
                & \HTwoBCodingGenes{}       & \HTwoBPseudoGenes{}
                & \HThreeCodingGenes{}      & \HThreePseudoGenes{}
                & \HFourCodingGenes{}       & \HFourPseudoGenes{}
                & \TotalCodingGenes{}       & \TotalPseudoGenes{} \\
        \bottomrule
      \end{tabular}
    \end{table*}

