\section{Histone genes}
  %% section flow: we start describing the organization very broadly. Then we actually get a
  %% clusters subsection, comparing between other organisms, and the distribution of genes
  %% in each one. Only after this is known, can we explain gene nomenclature.

  The genes encoding canonical histone isoforms are typically found in clusters in most metazoan genomes, 
  possibly for regulatory functions and to facilitate sequence conservation.
  %% Removed HLB in interests of focus and length

  \subsection{Histone gene clusters}
    The human genome contains \todo{total histone clusters} histone gene clusters 
	named as HIST1 to HIST\todo{total histone clusters} in order of decreasing histone gene count.
    The major and secondary histone gene clusters,
    HIST1 and HIST2 on chromosomes~\HISTOneLocus{} and \HISTTwoLocus{} respectively,
    contain contiguous high density arrays of histone genes. On the other hand, the minor
    cluster HIST3 on chromosome~\HISTThreeLocus{} has only a handful of them, while HIST4
    on \HISTFourLocus{} can barely be considered a cluster with its single gene
    (\tref{tab:histone-gene-count}).

    HIST1 contains \todo{percent genes in HIST1}\% of all histone genes, 
	encoding \CodingGenesInHISTOne{}~functional core histone genes plus all H1 histones. 
	It spans \HISTOneSpan{} and is the second most gene dense region at megabase scale in the genome \citep{MHC-III-analysis}. 
	The HIST1 location on chromosome~\HISTOneLocus{}
	is at the distal end of the major histocompatibility complex (MHC) in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999} 
	and it has been suggested that proximity to HIST1 can suppress recombinations in the MHC Class~I region \citep{MHC-repressed-by-HIST}.

    Although HIST2 contains only \CodingGenesInHISTTwo{}~coding genes it spans \HISTTwoSpan{} centred at \HISTTwoLocus{}.
	The genes are grouped in two sub-clusters with an extremely distant single pseudogene 
	and the coding gene density within the sub-culsters is comparable to HIST1. ???
	%% Is the sub-cluster density a correct statement?
    HIST2 at \HISTTwoLocus{} is prone to deletion and recurrent rearrangements \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}

	%% Removed HIST3 and HIST4 stuff because it doesn't say much
	%% Removed birth and death because it is not directly relevant to this data
	%% Removed Alu stuff because it has no biological implications
	%% Definitely need to keep these observation, maybe put in thesis discussion?

	%% By sea urchin do you mean L variegatus?
	Invertebrates histones such as \textit{Lytechinus variegatus} and \textit{Drosophila melanogaster} 
	occur in tens to hundreds of tandem repeats of units with one copy of the five histones.
	However, there is no apparent repetitive organisation of histones genes in the human genome 
	and the dispersive organisation suggests that histones expression is regulated by local elements \citep{close-regulators}.

  \subsection{Histone gene nomenclature}
    Histone genes have a specific nomenclature that encodes the cluster number and its
    position relative to the other histones \citep{Marzluff02}. An histone gene
    symbol, HIST1H3B for example, is divided into 3 parts: the cluster (HIST1),
    the histone type (H3), and its position (B). The position letter is interpreted as
    its position relative to the other histones of the same type, in the same cluster.
    Since the start of the cluster is defined as the side closer to the centromere, the
    example HIST1H3B translates to the second H3 of the histone cluster, the
    second closest to the centromere (\fref{fig:nomenclature}).

    \begin{figure*}
      \centering
      \includegraphics[width=\textwidth]{nomenclature-schematic.pdf}
      \caption{Histone gene nomenclature. a) canonical histone gene encode their position
               on the genome on their symbol. b) variant histone symbols are disperse through
               the genome and cannot be placed in relation to the others so their member
               letters derive from historical reasons.}
      \label{fig:nomenclature}
    \end{figure*}

    Two exceptions were made to this rule: 1) A missing position letter on the symbol indicates
    no other histones of the same type on the cluster. This is the case of HIST3H2A, the
    single H2A gene in HIST3. 2) There was no human HIST1H2AF. It exists an Hist1h2af mouse
    histone though, and this exception is likely to have been a compromise to accommodate
    matching the human symbols to the mouse orthologs, and ease the cross of data between the
    two model organisms.

    This was the original nomenclature as approved by the HUGO Gene Nomenclature Committee (HGNC)
    when the human genome was being finalized. However, new histones have since been
    reported (\tref{tab:difference-from-Marzluff02}) which forced new exceptions on the
    nomenclature to keep its apparent organization. For example, HIST2H2AA was renamed
    HIST2H2AA3 and a new HIST2H2AA4 has been added. Formerly, there was no distinction
    between the symbols of pseudo and coding genes, but new pseudo genes are suffixed with
    PS\# by order of discovery. But the ``old'' pseudo genes retain their original symbols
    so the lack of suffix does not indicate a coding protein.
    The end result of this nomenclature is that to avoid changing previously
    assigned gene symbols, the nomenclature lost its meaning. 

    The symbols for variant histones are much different and simpler. Due to their dispersion
    through the genome, no attempt was made to encode relative information on their name. In
    a similar fashion to H2AFX, symbols of histone variants are reduced to their histone type,
    followed by the letter F (standing for Family), and a set of characters that identified
    the variant, usually with historical or functional interpretation. For example, H2AFX and
    H2AFZ are named accordingly to their migration order in acetic acid-urea gels
    \citep{HTwoA-first-variants}.
    The only exception to this is the H3 variant CENP--A, since it is also the component A
    of the CENtromeric Protein, initially reported by \cite{CENPA-first-report} before being
    identified as an histone protein \citep{CENPA-copurifies-histones, CENPA-sequence-analysis}.



  \subsection{Gene promoters}
    \todo[inline]{We already have the 500 bp upstream of the TSS... we should
                do something with this data}

    Because histone genes have been so well conserved throughout the evolution, they make
    excellent models to study computational methods, even for promoters
    \citep{histone-promoter-modeling}.

    HIST1 is actually split into 2 parts, and different promoters have been reported for each
    part \citep{HISTOne-different-transcripts}.

    Automatic analysis of the genome after training with histone genes, identifies
    several other genes that are involved in the cell replication with points
    towards a common regulation method. They still lack the stem-loop though.

%% We don't really have anything to say about introns. Canonical histones
%% don't have them while variants might. We already said it on the introduction
%  \subsection{Introns}
%    Histone genes have no introns on their genes, probably to circumvent the
%    requirement for primary transcript processing when histones must be rapidly
%    produced at S~phase. Some variant histones have introns.

    \begin{table*}
      \caption{Count of expected coding and pseudo core histone genes,
               divided by histone cluster and histone type. The right-hand
               operand (marked with $\psi$) of each column corresponds to the
               number of pseudo genes.}
      \label{tab:histone-gene-count}
      \centering
      \begin{tabular}{l r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$}}
        \toprule
        \null   & \multicolumn{2}{c}{H2A}  & \multicolumn{2}{c}{H2B}
                & \multicolumn{2}{c}{H3}   & \multicolumn{2}{c}{H4}
                & \multicolumn{2}{c}{Total} \\
        \midrule
        HIST1   & \HTwoACodingInHISTOne{}     & \HTwoAPseudoInHISTOne{}
                & \HTwoBCodingInHISTOne{}     & \HTwoBPseudoInHISTOne{}
                & \HThreeCodingInHISTOne{}    & \HThreePseudoInHISTOne{}
                & \HFourCodingInHISTOne{}     & \HFourPseudoInHISTOne{}
                & \CodingGenesInHISTOne{}     & \PseudoGenesInHISTOne{} \\
        HIST2   & \HTwoACodingInHISTTwo{}     & \HTwoAPseudoInHISTTwo{}
                & \HTwoBCodingInHISTTwo{}     & \HTwoBPseudoInHISTTwo{}
                & \HThreeCodingInHISTTwo{}    & \HThreePseudoInHISTTwo{}
                & \HFourCodingInHISTTwo{}     & \HFourPseudoInHISTTwo{}
                & \CodingGenesInHISTTwo{}     & \PseudoGenesInHISTTwo{} \\
        HIST3   & \HTwoACodingInHISTThree{}   & \HTwoAPseudoInHISTThree{}
                & \HTwoBCodingInHISTThree{}   & \HTwoBPseudoInHISTThree{}
                & \HThreeCodingInHISTThree{}  & \HThreePseudoInHISTThree{}
                & \HFourCodingInHISTThree{}   & \HFourPseudoInHISTThree{}
                & \CodingGenesInHISTThree{}   & \PseudoGenesInHISTThree{} \\
        HIST4   & \HTwoACodingInHISTFour{}    & \HTwoAPseudoInHISTFour{}
                & \HTwoBCodingInHISTFour{}    & \HTwoBPseudoInHISTFour{}
                & \HThreeCodingInHISTFour{}   & \HThreePseudoInHISTFour{}
                & \HFourCodingInHISTFour{}    & \HFourPseudoInHISTFour{}
                & \CodingGenesInHISTFour{}  & \PseudoGenesInHISTFour{} \\
        \addlinespace
        Total   & \HTwoACodingGenes{}       & \HTwoAPseudoGenes{}
                & \HTwoBCodingGenes{}       & \HTwoBPseudoGenes{}
                & \HThreeCodingGenes{}      & \HThreePseudoGenes{}
                & \HFourCodingGenes{}       & \HFourPseudoGenes{}
                & \TotalCodingGenes{}       & \TotalPseudoGenes{} \\
        \bottomrule
      \end{tabular}
    \end{table*}

