\section{Histone genes}

  \subsection{Histone gene clusters}
	%% For simplicity, decided to list HIST4 explicitly for rather than allow it to come in by a variable
    The human genome contains histone gene clusters HIST1 to HIST4 named in order of decreasing histone gene count.
	The major and secondary histone gene clusters, 
	HIST1 on chromosome~\HISTOneLocus{} and HIST2 on chromosome~\HISTTwoLocus{},
	contain contiguous high density arrays of histone genes 
	whereas the minor cluster HIST3 on chromosome~\HISTThreeLocus{} has only a handful of genes (\tref{tab:histone-gene-count}).

    HIST1 contains \todo{percent genes in HIST1}\% of all histone genes, 
	encoding \CodingGenesInHISTOne{}~functional core histone genes plus all H1 histones. 
	It spans \HISTOneSpan{} \todo{need psece between number and unit}
	and is the second most gene dense region in the genome at megabase scale \citep{MHC-III-analysis}. 
	At \HISTOneLocus{}, HIST1 is located the distal end of the major histocompatibility complex (MHC) 
	in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999} 
	and it has been suggested that proximity to HIST1 can suppress recombinations in the MHC Class~I region \citep{MHC-repressed-by-HIST}.

    HIST2 contains only \CodingGenesInHISTTwo{}~coding genes 
	and is nominally less gene dense, spanning \HISTTwoSpan{} across \HISTTwoLocus{}.
	However, the histone genes in HIST2 coding functional proteins
	are grouped in two sub-clusters each of which has gene density is comparable to HIST1.
	A pseudogene idistant from the coding genes also inflates the apparent cluster span.
    HIST2 is reported to be prone to deletions and frequent rearrangements \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}.

	HIST3 and HIST4 contain only \CodingGenesInHISTThree{} and \CodingGenesInHISTFour{}~coding genes respectively. 
	Although the histone genes are tightly clustered in HIST1 and HIST2 there is no obvious repetitive organisation 
	and the dispersive organisation has been suggested to imply that their expression is regulated by local elements \citep{close-regulators}.

	%% Removed birth and death because it is not directly relevant to this data
	%% Removed Alu stuff because it has no biological implications
	%% Definitely need to keep these observation, maybe put in thesis discussion?

  \subsection{Histone gene nomenclature}
    Histone genes have a Human Genome Organisation (HUGO) Gene Nomenclature Committee (HGNC) naming system 
	based on encoding the cluster number and position relative to other histones \citep{Marzluff02}.
	%% Should we try putting a url reference into the manuscript?
	%% http://www.genenames.org/genefamilies/histones
	The official gene symbol is divided into 3 parts: the cluster, the histone type, and its positional order (\fref{fig:nomenclature}).
	The position letter is relative to the other histones of the same type in the same cluster, 
	and the orientation of the cluster is defined from the centromeric side. 
	For example, HIST1H2BD is the fourth closest H2B gene to the centromere in the HIST1 cluster.
	%% Changed example to match figure!
	
	In contrast histone variant genes are present in only one or two copies 
	that are dispersed across the genome outside clusters 
	so their nomenclature comprises the 3 parts: histone type, F for family, and an arbitrary historical identifier.
	For example, H2AFX is the gene name for H2AX.
	Where two isoforms exist the second has a number associated, 
	such as H2AY and H2AY2 for the two genes encoding isoforms of macroH2A.
	
    \begin{figure*}
      \centering
      \includegraphics[width=\textwidth]{nomenclature-schematic.pdf}
      \caption{Histone gene nomenclature. a) canonical histone gene encode their position
               on the genome on their symbol. b) variant histone symbols are disperse through
               the genome and cannot be placed in relation to the others so their member
               letters derive from historical reasons.}
      \label{fig:nomenclature}
    \end{figure*}

    Two exceptions were originally applied to these simple rules \citep{Marzluff02}: 
	Firstly, the position letter is omitted if there are no other histones of the same type in the cluster, 
	such as for HIST3H2A which is the sole H2A gene in HIST3. 
	Secondly, the human and mouse histone clusters are largely syntenic 
	so positional identifiers were skipped where one genome lacked a particular isoform 
	in order to maintain the equivalence of identifiers. 
	Thus there was no human HIST1H2AF but Hist1h2af exists in mouse.

	Unfortunately several new histone genes have been uncovered since the original naming (\tref{tab:difference-from-Marzluff02}). 
	This has forced additional nomenclature exceptions to maintain the apparent organization.
	For example, HIST2H2AA was renamed HIST2H2AA3 and a new HIST2H2AA4 has been added.
	%% What are HIST2H2AA1 and HIST2H2AA2?
	
	The use of a numerical identifier for the H3.3 variant causes an exception 
	where isoforms take on a distinguishing letter instead of a number, 
	thus producing H3F3A, H3F3B and H3F3C.
	
	For historical reasons, the H3 variant CENP--A takes its name as 
	component A of the CENtromeric Proteins \cite{CENPA-first-report} 
	and was only later identified as a histone protein \citep{CENPA-copurifies-histones, CENPA-sequence-analysis}.
	
	Furthermore, there was no distinction originally made between pseudogenes and functional coding genes, 
	so ??? is a pseudogene whereas its neighbour ??? is a coding gene.
	%% Give an example of adjacent genes where one is pseudo and one is functional
	Newly uncovered pseudogenes have since been suffixed with PS\# in order of discovery, such as HIST1H2APS2. 
	The ``original'' pseudogenes retain their symbols, so the lack of suffix does not indicate a functional gene.
	Furthermore, histone variant pseudogenes are suffixed instead with P\#, yielding symbols such as H2AFXP and H2AFZP5.

    These complex exceptions have caused the originally informative nomenclature to lose much of its meaning. 

%% Let's leave the promoters and introns because we don't have anything to say

%% This is arguably the most important table. We should cite it early and often!
    \begin{table*}
      \caption{Count of expected coding and pseudo core histone genes,
               divided by histone cluster and histone type. The right-hand
               operand (marked with $\psi$) of each column corresponds to the
               number of pseudo genes.}
      \label{tab:histone-gene-count}
      \centering
      \begin{tabular}{l r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$}}
        \toprule
        \null   & \multicolumn{2}{c}{H2A}  & \multicolumn{2}{c}{H2B}
                & \multicolumn{2}{c}{H3}   & \multicolumn{2}{c}{H4}
                & \multicolumn{2}{c}{Total} \\
        \midrule
        HIST1   & \HTwoACodingInHISTOne{}     & \HTwoAPseudoInHISTOne{}
                & \HTwoBCodingInHISTOne{}     & \HTwoBPseudoInHISTOne{}
                & \HThreeCodingInHISTOne{}    & \HThreePseudoInHISTOne{}
                & \HFourCodingInHISTOne{}     & \HFourPseudoInHISTOne{}
                & \CodingGenesInHISTOne{}     & \PseudoGenesInHISTOne{} \\
        HIST2   & \HTwoACodingInHISTTwo{}     & \HTwoAPseudoInHISTTwo{}
                & \HTwoBCodingInHISTTwo{}     & \HTwoBPseudoInHISTTwo{}
                & \HThreeCodingInHISTTwo{}    & \HThreePseudoInHISTTwo{}
                & \HFourCodingInHISTTwo{}     & \HFourPseudoInHISTTwo{}
                & \CodingGenesInHISTTwo{}     & \PseudoGenesInHISTTwo{} \\
        HIST3   & \HTwoACodingInHISTThree{}   & \HTwoAPseudoInHISTThree{}
                & \HTwoBCodingInHISTThree{}   & \HTwoBPseudoInHISTThree{}
                & \HThreeCodingInHISTThree{}  & \HThreePseudoInHISTThree{}
                & \HFourCodingInHISTThree{}   & \HFourPseudoInHISTThree{}
                & \CodingGenesInHISTThree{}   & \PseudoGenesInHISTThree{} \\
        HIST4   & \HTwoACodingInHISTFour{}    & \HTwoAPseudoInHISTFour{}
                & \HTwoBCodingInHISTFour{}    & \HTwoBPseudoInHISTFour{}
                & \HThreeCodingInHISTFour{}   & \HThreePseudoInHISTFour{}
                & \HFourCodingInHISTFour{}    & \HFourPseudoInHISTFour{}
                & \CodingGenesInHISTFour{}  & \PseudoGenesInHISTFour{} \\
        \addlinespace
        Total   & \HTwoACodingGenes{}       & \HTwoAPseudoGenes{}
                & \HTwoBCodingGenes{}       & \HTwoBPseudoGenes{}
                & \HThreeCodingGenes{}      & \HThreePseudoGenes{}
                & \HFourCodingGenes{}       & \HFourPseudoGenes{}
                & \TotalCodingGenes{}       & \TotalPseudoGenes{} \\
        \bottomrule
      \end{tabular}
    \end{table*}

