\section{Histone genes}

  \subsection{Histone gene clusters}
	%% For simplicity, decided to list HIST4 explicitly for rather than allow it to come in by a variable
    The human canonical histone genes are located in clusters HIST1 to HIST4, 
	named in order of decreasing histone gene count (\tref{tab:histone-gene-count}).
	The major histone gene cluster HIST1 at locus~\HISTOneLocus{}  
	contains \todo{percent genes in HIST1}\% of all histone genes 
	comprising \CodingGenesInHISTOne{}~functional core histone genes plus all canonical H1 histones. 
	HIST2 at locus~\HISTTwoLocus{} contains \CodingGenesInHISTTwo{}~coding genes. 
	HIST3 at \HISTThreeLocus{} contains \CodingGenesInHISTThree{} coding genes 
	and HIST4 at \HISTFourLocus{} contains only \CodingGenesInHISTFour{}~coding gene. 
	Histone variants are dispersed across the genome outside the clusters, 
	so for example the three H3.3 variant encoding genes are located on chromosomes 1, 12 and 17 
	far removed from other histone genes.

	HIST1 and HIST2 are both contiguous high density arrays of histone genes. 
	HIST1 spans \HISTOneSpan{} \todo{need space between number and unit}
	and is the second most gene dense region in the human genome at megabase scale \citep{MHC-III-analysis}.

	However, progress in understanding histone gene function suggests 
	that histone cluster interpretation may change. 
	For example, although \textit{HIST1H2BA} is nominally a canonical histone 
	it has recently been highlighted that this is testes-specific TH2 \addref{UruhamaKurumizakaACTAcrystF2014}.
	This H2B gene encodes the most divergent canonical H2B protein 
	and is located 300 kbp upstream of the rest of the HIST1 cluster 
	separated by a number of non-histone genes. 
	\textit{HIST1H2BA} is adjacent and divergently transcribed from \textit{HIST1H2AA} 
	which encodes a protein of similarly high variation from the most common canonical H2A isoform. 
	One possibility is that this gene pair are histone variants 
	located adjacent to the HIST1 cluster that illustrate histone sub-functionalisation.

	If \textit{HIST1H2AA} and \textit{HIST1H2BA} are excluded, 
	the hemochromatosis protein encoding gene \textit{HFE} \addref{AlbigDoeneckeJCB1998} 
	is the sole non-histone protein coding gene located within any of the human histone gene clusters. 

	The implications of this tight and exclusive histone gene clustering 
	is not well characterised \addref{Eirin-Lopez EvBiolCh8 2009} 
	but has been suggested to imply that expression is regulated by local elements \citep{close-regulators}.
	
	An alternative functional explanation for clustering as a mechanism for 
	histone gene sequence conservation is disputed \addref{NeiRooneyARG2005}.
	HIST1 is located towards the distal end of the major histocompatibility complex (MHC) 
	in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999}, 
	and it has been suggested that proximity to HIST1 
	suppresses recombination in the MHC Class~I region \citep{MHC-repressed-by-HIST}. 
	In contrast it has been reported that HIST2 is prone to 
	deletions and frequent rearrangements \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}. 

  \subsection{Canonical histone gene nomenclature}
    Canonical histone gene names adhere to a Human Genome Organisation (HUGO) Gene Nomenclature Committee (HGNC) 
	endorsed system derived from the cluster number and position relative to other histones \citep{Marzluff02}.
	%% Should we try putting a url reference into the manuscript?
	%% http://www.genenames.org/genefamilies/histones

	The human canonical histone gene symbols are divided into 3 parts: 
	HIST cluster, histone type, and identifier letter
	for the order relative to other histone genes of the same type in the cluster (\fref{fig:nomenclature}A).
	For example, \textit{HIST1H2BD} is nominally the fourth H2B gene in the HIST1 cluster.
	The direction of the cluster order is arbitrary, 
	so HIST1 is ordered from the centromeric side 
	whereas HIST2 and HIST3 are ordered from the telomeric side.
	%% Think this is correct from viewing the genome?
	
	Two exceptions were originally applied to this simple gene naming rule \citep{Marzluff02}. 
	Firstly, the position identifier is omitted if there are no other histones of the same type in the cluster, 
	so \textit{HIST3H2A} is the sole H2A gene in HIST3. 
	Secondly, the human and mouse histone clusters are largely syntenic 
	so positional identifiers for missing orthologues were skipped to maintain the equivalence of gene symbols.
	This means there is no human \textit{HIST1H2AF} although \textit{Hist1h2af} exists in mouse.

	Several new histone genes have been uncovered since the original naming (\tref{tab:difference-from-Marzluff02}) 
	and this required additional nomenclature exceptions.
	For example, new H2A encoding genes identified in the human and mouse HIST2 cluster after \textit{HIST2H2AA} 
	required renaming of \textit{HIST2H2AA} to \textit{HIST2H2AA3} 
	and the addition of a new human gene as \textit{HIST2H2AA4}. 
	There are no human orthologues of mouse \textit{Hist2h2aa1} and \textit{Hist2h2aa2}.
	
	Furthermore, no distinction was originally made between pseudogenes and functional coding genes, 
	so \textit{HIST3H2AA} is a pseudogene whereas neighbouring \textit{HIST3H2AB} is a coding gene.
	Newly uncovered pseudogenes are now suffixed with PS and a number in order of discovery, 
	such as \textit{HIST1H2APS5}. 
	Nevertheless, the pseudogenes named in teh origianl classification retain their symbols without PS. 
	This means that the absence of a PS suffix does not indicate a functional gene, 
	and that there is no positional information in the gene symbols of new pseudogenes.

  \subsection{Histone variant gene nomenclature}
	
	All recognised histone variant gene families comprise only one or a few copies 
	dispersed across the genome outside clusters.
	The histone variant gene symbols have a separate 3 part nomenclature 
	consisting of labels for histone type, F for family, 
	and an identifier letter based on discovery order (\fref{fig:nomenclature}B).
	Where two or more isoforms exist the copies take a numerical suffix, 
	although this is sometimes omitted for the first copy. 
	For example, \textit{H2AY} and \textit{H2AY2} encode the human macroH2As, 
	with \textit{H2AY} giving rise to alternative transcripts encoding macroH2A1.1 and macroH2A1.2 protein isoforms 
	whereas \textit{H2AY2} encodes macroH2A2.
	
	The variant histone gene symbols (\tref{tab:histone-variant-families}) also contain exceptions.
	For example, the three gene copies encoding the H3.3 variant 
	take on distinguishing letters \textit{H3F3A}, \textit{H3F3B} and \textit{H3F3C} instead of numbers. 
	The H3 variant CENPA derives its name as the first identified centromeric protein \cite{CENPA-first-report} 
	before it was realised to be a histone \citep{CENPA-copurifies-histones, CENPA-sequence-analysis}.
	
	Histone variant pseudogenes are suffixed with P to yield symbols such as \textit{H2AFXP} and \textit{H2AFZP5}.
	Increasing interest in histone variant function \addref{Maze Allis NRG 2014}
	coupled with a variety of usages and conflicts between species 
	has led to guidelines for improved consistency in variant histone nomenclature \addref{Talbert Epigen Chrom 2012}.

  \subsection{Histone gene sequences}

	The alignment of coding regions for functional canonical histone genes 
	demonstrates considerable sequence variation at the DNA level (\fref{fig:histone-gene-variation}). 
	This relatively high sequence variation in human canonical histone genes 
	is consistent with observations that 
	synonymous codon divergence exceeds non-synonymous variation 
	across eukaryotes \addref{Piontkivska MolBiolEvol 2000} \addref{Rooney MolBiolEvol 2002}. 
	It supports the hypothesis that histone protein sequence conservation 
	results from birth and death evolution through strong selective pressure at the protein level, 
	and not from a homogenisation mechanism at the gene level based on clustering \addref{NeiRooney AnnRevGen 2005}.

    \begin{figure*}
      \centering
      \includegraphics[width=\textwidth]{nomenclature-schematic.pdf}
      \caption{Histone gene nomenclature. 
	  A. Canonical histone gene names encode their relative genomic order by cluster. 
	  \todo[inline]{Parts A and B need labels. Gene names so need to be italics!}
	  B. Variant histone symbols are identified with F then a family identifier letter
	  and numbered by discovery, except 1 is omitted for the founder member.
	  \todo[inline]{Change part B to use H2AFY2 as a more complete example of variant nomenclature}
	  H3.3 genes conversely use numbers and letters for family and member.
	  C. Canonical pseudogenes identified since 2002 are named by cluster with PS identifier numbered by discovery. 
	  Previously canonical pseudogenes were named like functional histone genes.}
	  \todo[inline]{Include part C using HIST1H2APS2 as example of PS pseudogene nomenclature}
      \label{fig:nomenclature}
    \end{figure*}

  \begin{table*}
    \caption{HGNC recognised histone variant family stem names, commonly used protein names 
	and names for improved consistency based on proposal by Talbert et al (2012). 
	HIST1H2BA is nominally not a variant but is included for completeness.}
    \label{tab:histone-variant-families}
    \centering
    \begin{tabular}{l l l l}
      \toprule
      Family & Protein name & Talbert et al name & Notes \\
      \midrule
	  H2AFB & H2A.Bbd & H2A.B & equivalent to H2A.L \\
	  H2AFJ & H2A.J & H2A.J & at HIST4 locus \\
	  H2AFV & H2A.Z-2 & H2A.Z.1 & - \\
	  H2AFX & H2AX & H2A.X & - \\
	  H2AFY & macroH2A/mH2A & macroH2A & - \\
	  H2AFZ & H2A.Z & H2A.Z.1 & - \\
	  H2BFM & ? & H2B.M & homologous to H2B.W, X-linked\\
	  H2BFS & - & - & pseudogene \\
	  H2BFWT & ? & H2B.W & testes specific, X-linked \\
	  H2BFX & - & - & pseudogene \\
	  HIST1H2BA & TH2B/TSH2B & H2B.1 & testes specific \\
	  H3F3 & H3.3 & H3.3 & euchromatin related \\
	  CENPA & CENP-A & ? & centromere-specific \\
	\bottomrule
    \end{tabular}
  \end{table*}

%% This is arguably the most important table. We should cite it as early as possible
    \begin{table*}
      \caption{
	  Count of human canonical histone coding and pseudogenes by histone cluster and type.
	  The right-hand value in each cell marked with $\psi$ is the number of pseudo genes.}
      \label{tab:histone-gene-count}
      \centering
      \begin{tabular}{l r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$}}
        \toprule
        \null   & \multicolumn{2}{c}{H2A}  & \multicolumn{2}{c}{H2B}
                & \multicolumn{2}{c}{H3}   & \multicolumn{2}{c}{H4}
                & \multicolumn{2}{c}{Total} \\
        \midrule
        HIST1   & \HTwoACodingInHISTOne{}     & \HTwoAPseudoInHISTOne{}
                & \HTwoBCodingInHISTOne{}     & \HTwoBPseudoInHISTOne{}
                & \HThreeCodingInHISTOne{}    & \HThreePseudoInHISTOne{}
                & \HFourCodingInHISTOne{}     & \HFourPseudoInHISTOne{}
                & \CodingGenesInHISTOne{}     & \PseudoGenesInHISTOne{} \\
        HIST2   & \HTwoACodingInHISTTwo{}     & \HTwoAPseudoInHISTTwo{}
                & \HTwoBCodingInHISTTwo{}     & \HTwoBPseudoInHISTTwo{}
                & \HThreeCodingInHISTTwo{}    & \HThreePseudoInHISTTwo{}
                & \HFourCodingInHISTTwo{}     & \HFourPseudoInHISTTwo{}
                & \CodingGenesInHISTTwo{}     & \PseudoGenesInHISTTwo{} \\
        HIST3   & \HTwoACodingInHISTThree{}   & \HTwoAPseudoInHISTThree{}
                & \HTwoBCodingInHISTThree{}   & \HTwoBPseudoInHISTThree{}
                & \HThreeCodingInHISTThree{}  & \HThreePseudoInHISTThree{}
                & \HFourCodingInHISTThree{}   & \HFourPseudoInHISTThree{}
                & \CodingGenesInHISTThree{}   & \PseudoGenesInHISTThree{} \\
        HIST4   & \HTwoACodingInHISTFour{}    & \HTwoAPseudoInHISTFour{}
                & \HTwoBCodingInHISTFour{}    & \HTwoBPseudoInHISTFour{}
                & \HThreeCodingInHISTFour{}   & \HThreePseudoInHISTFour{}
                & \HFourCodingInHISTFour{}    & \HFourPseudoInHISTFour{}
                & \CodingGenesInHISTFour{}  & \PseudoGenesInHISTFour{} \\
        \addlinespace
        Total   & \HTwoACodingGenes{}       & \HTwoAPseudoGenes{}
                & \HTwoBCodingGenes{}       & \HTwoBPseudoGenes{}
                & \HThreeCodingGenes{}      & \HThreePseudoGenes{}
                & \HFourCodingGenes{}       & \HFourPseudoGenes{}
                & \TotalCodingGenes{}       & \TotalPseudoGenes{} \\
        \bottomrule
      \end{tabular}
    \end{table*}

    \begin{figure*}
      \centering
      \subtop[H2A]{
        \includegraphics[width=0.48\textwidth]
        {seqlogo_H2A_cds.pdf}
        \label{fig:H2A-cds-weblogo}
      }
      \hfill
      \subtop[H2B]{
        \includegraphics[width=0.48\textwidth]
        {seqlogo_H2B_cds.pdf}
        \label{fig:H2B-cds-weblogo}
      }
      \subtop[H3]{
        \includegraphics[width=0.48\textwidth]
        {seqlogo_H3_cds.pdf}
        \label{fig:H3-cds-weblogo}
      }
      \hfill
      \subtop[H4]{
        \includegraphics[width=0.48\textwidth]
        {seqlogo_H4_cds.pdf}
        \label{fig:H4-cds-weblogo}
      }
      \captionof{figure}{Sequence logos for canonical coding histone genes listed in table XX. 
                         Coding regions of A. H2A, B. H2B, C. H3 and D. H4.}
      \label{fig:histone-gene-variation}
    \end{figure*}

