\section{Histone genes}
  %% section flow: we start describing the organization very broadly. Then we actually get a
  %% clusters subsection, comparing between other organisms, and the distribution of genes
  %% in each one. Only after this is known, can we explain gene nomenclature.

  The genes encoding canonical histone isoforms are found in clusters in most metazoan genomes, 
  possibly for regulatory functions and to facilitate sequence conservation.
  %% Removed HLB in interests of focus and length

  \subsection{Histone gene clusters}
    The human genome contains \todo{total histone clusters} histone gene clusters 
	named as HIST1 to HIST\todo{total histone clusters} in order of decreasing histone gene count.
    The major and secondary histone gene clusters,
    HIST1 on chromosome~\HISTOneLocus{} and HIST2 on chromosome~\HISTTwoLocus{},
    contain contiguous high density arrays of histone genes.
	The minor cluster HIST3 on chromosome~\HISTThreeLocus{} has only a handful of genes.
    (\tref{tab:histone-gene-count}).

    HIST1 contains \todo{percent genes in HIST1}\% of all histone genes, 
	encoding \CodingGenesInHISTOne{}~functional core histone genes plus all H1 histones. 
	It spans \HISTOneSpan{} and is the second most gene dense region in the genome at megabase scale \citep{MHC-III-analysis}. 
	The location of HIST1 on chromosome~\HISTOneLocus{}
	puts it at the distal end of the major histocompatibility complex (MHC) in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999} 
	and it has been suggested that proximity to HIST1 can suppress recombinations in the MHC Class~I region \citep{MHC-repressed-by-HIST}.

    Although HIST2 contains only \CodingGenesInHISTTwo{}~coding genes it spans \HISTTwoSpan{} centred at \HISTTwoLocus{}.
	The histone genes in HIST2 are grouped in two sub-clusters 
	within which the coding gene density is comparable to HIST1.
	%% Is the sub-cluster density a correct statement?
    HIST2 is prone to deletion and recurrent rearrangements \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}

	%% Removed HIST3 and HIST4 stuff because it doesn't say much
	%% Removed birth and death because it is not directly relevant to this data
	%% Removed Alu stuff because it has no biological implications
	%% Definitely need to keep these observation, maybe put in thesis discussion?
	There is no apparent repetitive organisation of histones genes in the human genome 
	and the dispersive organisation suggests that histones expression is regulated by local elements \citep{close-regulators}.
	%% By sea urchin do you mean L variegatus?
	In contrast, invertebrates histones such as \textit{Lytechinus variegatus} and \textit{Drosophila melanogaster} 
	occur in tens to hundreds of tandem repeating units comprising a copy of each of the five histones.

  \subsection{Histone gene nomenclature}
    Histone genes have a HUGO Gene Nomenclature Committee (HGNC) nomenclature 
	that encodes the cluster number and its position relative to the other histones \citep{Marzluff02}.
	A gene symbol is divided into 3 parts: the cluster, the histone type, and its positional order.
	The position letter is relative to the other histones of the same type, in the same cluster.
    The orientation of the cluster is defined from the centromeric side, 
	so HIST1H3B is the second closest H3 gene to teh ecntromere in the HIST1 cluster (\fref{fig:nomenclature}).

    \begin{figure*}
      \centering
      \includegraphics[width=\textwidth]{nomenclature-schematic.pdf}
      \caption{Histone gene nomenclature. a) canonical histone gene encode their position
               on the genome on their symbol. b) variant histone symbols are disperse through
               the genome and cannot be placed in relation to the others so their member
               letters derive from historical reasons.}
      \label{fig:nomenclature}
    \end{figure*}

    Two exceptions were originally applied to these simple rules: 
	1) A missing position letter on the symbol indicates no other histones of the same type on the cluster. 
	This is the case of HIST3H2A, the single H2A gene in HIST3. 
	2) There was no human HIST1H2AF although Hist1h2af exists in mouse. This exception maintains synteny so that
    human and mouse orthologs have equivalnet symbols to ease comparisons.

	Unfortunatley the uncovering of new histone genes since the original naming (\tref{tab:difference-from-Marzluff02}) 
	has forced additional nomenclature exceptions to maintain the apparent organization.
	For example, HIST2H2AA was renamed HIST2H2AA3 and a new HIST2H2AA4 has been added.
	%% What are HIST2H2AA1 and HIST2H2AA2?
	
	In addition there was no distinction between the symbols of pseudo and coding genes, 
	but new pseudo genes have since been suffixed with PS\# by order of discovery. 
	Nevertheless the ``old'' pseudo genes retain their original symbols 
	so the lack of suffix does not indicate a coding protein.

    Although gene names have been maintained, these complex exceptions have caused the nomenclature 
	to lose much of its location meaning. 

    The gene symbols for variant histones are different and simpler. 
	Since they are disributed through the genome without clustering, 
	no attempt was made to encode positional information on their name.
	Histone variant gene nomenclature is based on the histone type,
    followed by the letter F (standing for Family), 
	and a set of characters that identify the variant on historical or functional grounds. 
	For example, H2AFX and H2AFZ are named accordingly to their migration in acetic acid-urea gels
    \citep{HTwoA-first-variants}.
    The only exception to this is the H3 variant CENP--A, 
	because it is also the component A of the CENtromeric Proteins
	 \cite{CENPA-first-report} and was only later identified as a histone protein \citep{CENPA-copurifies-histones, CENPA-sequence-analysis}.

  \subsection{Gene promoters}
    \todo[inline]{We already have the 500 bp upstream of the TSS... we should
                do something with this data}

    Because histone genes have been so well conserved throughout the evolution, they make
    excellent models to study computational methods, even for promoters
    \citep{histone-promoter-modeling}.

    HIST1 is actually split into 2 parts, and different promoters have been reported for each
    part \citep{HISTOne-different-transcripts}.

    Automatic analysis of the genome after training with histone genes, identifies
    several other genes that are involved in the cell replication with points
    towards a common regulation method. They still lack the stem-loop though.

%% We don't really have anything to say about introns. Canonical histones
%% don't have them while variants might. We already said it on the introduction
%  \subsection{Introns}
%    Histone genes have no introns on their genes, probably to circumvent the
%    requirement for primary transcript processing when histones must be rapidly
%    produced at S~phase. Some variant histones have introns.

    \begin{table*}
      \caption{Count of expected coding and pseudo core histone genes,
               divided by histone cluster and histone type. The right-hand
               operand (marked with $\psi$) of each column corresponds to the
               number of pseudo genes.}
      \label{tab:histone-gene-count}
      \centering
      \begin{tabular}{l r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$} r@{ + }l<{$\psi$}}
        \toprule
        \null   & \multicolumn{2}{c}{H2A}  & \multicolumn{2}{c}{H2B}
                & \multicolumn{2}{c}{H3}   & \multicolumn{2}{c}{H4}
                & \multicolumn{2}{c}{Total} \\
        \midrule
        HIST1   & \HTwoACodingInHISTOne{}     & \HTwoAPseudoInHISTOne{}
                & \HTwoBCodingInHISTOne{}     & \HTwoBPseudoInHISTOne{}
                & \HThreeCodingInHISTOne{}    & \HThreePseudoInHISTOne{}
                & \HFourCodingInHISTOne{}     & \HFourPseudoInHISTOne{}
                & \CodingGenesInHISTOne{}     & \PseudoGenesInHISTOne{} \\
        HIST2   & \HTwoACodingInHISTTwo{}     & \HTwoAPseudoInHISTTwo{}
                & \HTwoBCodingInHISTTwo{}     & \HTwoBPseudoInHISTTwo{}
                & \HThreeCodingInHISTTwo{}    & \HThreePseudoInHISTTwo{}
                & \HFourCodingInHISTTwo{}     & \HFourPseudoInHISTTwo{}
                & \CodingGenesInHISTTwo{}     & \PseudoGenesInHISTTwo{} \\
        HIST3   & \HTwoACodingInHISTThree{}   & \HTwoAPseudoInHISTThree{}
                & \HTwoBCodingInHISTThree{}   & \HTwoBPseudoInHISTThree{}
                & \HThreeCodingInHISTThree{}  & \HThreePseudoInHISTThree{}
                & \HFourCodingInHISTThree{}   & \HFourPseudoInHISTThree{}
                & \CodingGenesInHISTThree{}   & \PseudoGenesInHISTThree{} \\
        HIST4   & \HTwoACodingInHISTFour{}    & \HTwoAPseudoInHISTFour{}
                & \HTwoBCodingInHISTFour{}    & \HTwoBPseudoInHISTFour{}
                & \HThreeCodingInHISTFour{}   & \HThreePseudoInHISTFour{}
                & \HFourCodingInHISTFour{}    & \HFourPseudoInHISTFour{}
                & \CodingGenesInHISTFour{}  & \PseudoGenesInHISTFour{} \\
        \addlinespace
        Total   & \HTwoACodingGenes{}       & \HTwoAPseudoGenes{}
                & \HTwoBCodingGenes{}       & \HTwoBPseudoGenes{}
                & \HThreeCodingGenes{}      & \HThreePseudoGenes{}
                & \HFourCodingGenes{}       & \HFourPseudoGenes{}
                & \TotalCodingGenes{}       & \TotalPseudoGenes{} \\
        \bottomrule
      \end{tabular}
    \end{table*}

