\section{Histone genes}

  \subsection{Canonical core histone gene nomenclature}
    Canonical core histone genes adhere to a Human Genome Organisation (HUGO)
    Gene Nomenclature Committee (HGNC)
    endorsed system derived from the cluster number and position relative
    to other histones \citep{Marzluff02}.
    This superseded an earlier arbitrary scheme with backslashes (e.g. H2b/b)
    that preceded genome sequencing \citep{AlbigGenomics1997,AlbigHumangen1997}.

    The canonical core histone gene symbols are divided into 3 parts:
    HIST cluster, histone type, and identifier letter
    for the order relative to other histone genes of the same type in
    the cluster (\fref{subfig:nomenclature-coding-canonical}).
    For example, \textit{HIST1H2BD} is nominally the fourth H2B coding gene in the HIST1 cluster.
    Identifiers are ordered by their genomic coordinates starting at
    the telomere of the short arm \citep{Marzluff02}.

    \begin{figure*}
      \centering
      \subbottom[canonical coding gene]{%
        \begin{minipage}{0.37\textwidth}
          \centering
          \Huge{%
            \colorbox{red}{\strut HIST1}%
            \colorbox{green}{\strut H2B}%
            \colorbox{blue!40}{\strut D}%
          }

          \scriptsize{%
            \colorbox{blue!40}{\strut 4\textsuperscript{th}}
            \colorbox{green}{\strut H2B}
            \colorbox{red}{\strut in Histone cluster 1}
          }
        \end{minipage}
        \label{subfig:nomenclature-coding-canonical}
      }
      \subbottom[canonical pseudogene]{%
        \begin{minipage}{0.37\textwidth}
          \centering
          \Huge{%
            \colorbox{red}{\strut HIST2}%
            \colorbox{green}{\strut H3}%
            \colorbox{yellow}{\strut PS}%
            \colorbox{blue!40}{\strut 2}%
          }

          \scriptsize{%
            \colorbox{blue!43}{\strut 2\textsuperscript{nd}}
            \colorbox{green}{\strut H3}
            \colorbox{yellow}{\strut pseudogene}
            \colorbox{blue!40}{\strut found}
            \colorbox{red}{\strut in cluster 2}
          }
        \end{minipage}
        \label{subfig:nomenclature-pseudo-canonical}
      }
      \subbottom[variant]{%
        \begin{minipage}{0.22\textwidth}
          \centering
          \Huge{%
            \colorbox{red}{\strut H2A}%
            \colorbox{green}{\strut F}%
            \colorbox{blue!40}{\strut X}%
          }

          \scriptsize{%
            \colorbox{red}{\strut H2A}
            \colorbox{green}{\strut Family}
            \colorbox{blue!40}{\strut member X}
          }
        \end{minipage}
        \label{subfig:nomenclature-variant}
      }
      \caption{Histone gene nomenclature.
               \subcaptionref{subfig:nomenclature-coding-canonical}
               Canonical core histone gene names encode relative genomic
               order by cluster.
               \subcaptionref{subfig:nomenclature-pseudo-canonical}
               Pseudogenes named since 2002 include cluster, PS label,
               and discovery order identifier.
               \subcaptionref{subfig:nomenclature-variant}
               Most variant core histone genes are identified by type
               then F for family and identifier letter.}
    \end{figure*}

    Two exceptions were originally applied to these simple naming rules \citep{Marzluff02}.
    Firstly, the positional identifier is omitted if there are no
    other histones of the same type in the cluster,
    so \textit{HIST3H2A} is the sole H2A gene in HIST3.
    Secondly, the human and mouse histone clusters are largely syntenic
    so positional identifier letters for missing orthologs were skipped
    to maintain the equivalence of gene symbols.
    Consequently there is no human \textit{HIST1H2AF} to accommodate
    \textit{Hist1h2af} in mouse while keeping both --E and --G identifiers
    consistent for mouse and human orthologs.

    Several new histone genes have been uncovered since the original
    naming (\tref{tab:difference-from-Marzluff02})
    and this required additional nomenclature exceptions.
    For example, new H2A encoding genes were identified in both human
    and mouse HIST2 cluster preceding \textit{HIST2H2AA}
    leading to the renaming of \textit{HIST2H2AA} to \textit{HIST2H2AA3}
    and the addition of a new human gene as \textit{HIST2H2AA4}.
    There are no human orthologs of mouse \textit{Hist2h2aa1} and \textit{Hist2h2aa2}.

    Furthermore, no distinction was originally made between
    pseudogenes and functional coding genes,
    so \textit{HIST3H2BA} is a pseudogene whereas neighbouring
    \textit{HIST3H2BB} is the only functional H2B coding gene in HIST3.
    The HGNC definition of a pseudogene is
    a sequence that is generally untranscribed and untranslated
    but which has at least 50\% predicted amino acid identity
    across 50\% of the open reading frame to a named gene \citep{HGNC2013}.
    Newly uncovered histone pseudogenes are now suffixed with PS and a
    number in order of discovery (\fref{subfig:nomenclature-pseudo-canonical}),
    such as \textit{HIST1H2APS5} as the fifth H2A pseudogene discovered in HIST1.
    However, the pseudogenes named in the original classification retain their symbols without PS.
    This means that the absence of a PS suffix does not indicate a functional gene,
    and that there is no positional information in the gene symbols of most pseudogenes.

  \subsection{Histone gene clustering}
    The human canonical core histone genes are located in clusters HIST1 to HIST\NumberOfClusters{},
    named in order of decreasing histone gene count (\tref{tab:histone-gene-count}).
    HIST1 is the major histone gene cluster at locus~\HISTOneLocus{}
    with \CoreCodingGenesInHISTOne{}~functional core histone genes plus all canonical H1 histones,
    representing \FPround{\result}{\PercentageGenesInHISTOne}{0} \SI{\result}{\percent}
    of all canonical core histone genes.
    HIST2 at locus \HISTTwoLocus{} contains \CoreCodingGenesInHISTTwo{}~coding genes,
    HIST3 at locus \HISTThreeLocus{} contains \CoreCodingGenesInHISTThree{}~coding genes,
    and HIST4 at locus \HISTFourLocus{} contains \CoreCodingGenesInHISTFour{}~coding gene.

    HIST1 and HIST2 are both contiguous high density arrays of histone genes.
    HIST1 spans \AutoSIPrefix{\HISTOneSpan}{1}{\bp}
    and is the second most gene dense region in the human genome at
    megabase scale after the MHC class III region \citep{MHC-III-analysis}.
    The only non-histone protein coding gene located within the principal region of this cluster
    is \textit{HFE}, encoding the hemochromatosis protein \citep{AlbigDoenecke1998},
    although a number of other genes are located
    proximal to the outlying \textit{HIST1H2AA} and \textit{HIST1H2BA} pair.

    It has also been argued that histone clustering
    does not contribute to gene conversion \citep{NeiRooney2005}.
    HIST1 is located towards the distal end of the major histocompatibility complex (MHC)
    in the extended class~I region \citep{MHC-I-transcript, MHC-complete-sequencing-1999}
    and it has even been suggested that this proximity
    may suppresses recombination \citep{MHC-repressed-by-HIST}.
    In contrast, HIST2 may be prone to deletions and frequent rearrangements
    \citep{HISTTwo-prone-deletion-discovery, HISTTwo-prone-deletion-focus}

    The functional significance of histone gene clustering remains to be demonstrated.
    It has been suggested that clustering may facilitate coordinate regulation \citep{Eirinlopez2009,close-regulators},
    but interpreting such a functional relationship with genome organisation
    requires an accurate catalogue of the histone genes and their individual roles.

    Conversely, progress in understanding histone gene function
    suggests there is a need to update the definitions of the canonical histone gene clusters.
    For example, the \textit{HIST1H2AA} and \textit{HIST1H2BA} genes
    are located \SI{300}{\kilo\bp} upstream of the rest of the HIST1 cluster
    and separated from other cluster members a number of non-histone genes.
    Although the two genes have been assigned canonical histone gene names,
    the \textit{HIST1H2BA} gene in fact encodes the most divergent canonical H2B protein
    which is also known as TH2, TH2B, or hTSH2B and considered to be
    a testes-specific histone protein \citep{Zalensky2002,LiAusio2005,Shinagawa2014}.
    Immediately adjacent is \textit{HIST1H2AA} encoding
    a H2A protein isoform of similarly high variation.
    The syntenic rat orthologues of \textit{HIST1H2AA} and \textit{HIST1H2BA}
    are divergently transcribed specifically in testes \citep{HuhChae1991},
    and the mouse orthologues H2AL1/TH2a and TH2B have been shown to participate in gametogenesis \citep{GovinCaron2007}
    as well as to enhance stem cell reprogramming \citep{ShinagawaIshii2014,PadavattanKumarevel2015}.
    It is therefore possible that \textit{HIST1H2AA} and \textit{HIST1H2BA}
    are undergoing sub-functionalisation and could be reclassified
    as histone variants H2A.L and H2B.1 \citep{Talbert2012}.
    This would in turn lead to a recalculation of the length of the HIST1 cluster.

    A similar case may exist for the small HIST3 cluster \SI{80}{\mega\bp} downstream of HIST2.
    HIST3 contains protein coding genes \textit{HIST3H2A}, \textit{HIST3H2BB}, and \textit{HIST3H3}.
    \textit{HIST3H3} encodes testes-specific H3T/H3.4
    that has distinctive biochemical properties \citep{WittExpCellRes1996,KurumizakaCOSB2013},
    while the proteins encoded by \textit{HIST3H2A} and \textit{HIST3H2BB}
    are amongst the most divergent canonical histones of their types.

    The examples of the \textit{HIST1H2AA} and \textit{HIST1H2BA} pair
    and the HIST3 cluster illustrate
    the evolving nature of the human canonical histone complement
    and the need for a dynamic approach to classification.

  \subsection{Histone gene sequences}
    Despite the high conservation of canonical core histone protein sequences,
    the coding regions of these genes exhibit considerable variation
    (\fref{fig:h2a-histone-gene-variation},~\ref{fig:h2b-histone-gene-variation},
    ~\ref{fig:h3-histone-gene-variation}, and~\ref{fig:h4-histone-gene-variation}).
    These differences are largely located in the third base position of codons
    and reflect the very high ratio of synonymous to non-synonymous
    substitutions (\tref{tab:histone-gene-differences}).

    The mean number of synonymous substitutions per site ($d_S$)
    in the sets of histone gene type isoforms is
    \FPround{\result}{\MeanHTwoAdS}{1} \result{} for H2A,
    \FPround{\result}{\MeanHTwoBdS}{1} \result{} for H2B,
    \FPround{\result}{\MeanHThreedS}{1} \result{} for H3,
    and \FPround{\result}{\MeanHFourdS}{1} \result{} for H4.
    This is consistent with observations that
    synonymous codon divergence far exceeds non-synonymous variation
    for histone genes across eukaryotes \citep{Piontkivska2002, Rooney2002}.
    It supports a hypothesis that histone protein sequence conservation
    results from birth and death evolution through strong selective pressure
    at the protein level rather than
    sequence homogenisation based on clustering \citep{NeiRooney2005}.

    Despite the level of synonymous substitution,
    codon usage is strongly biased towards the most frequently used
    human codons for most amino acids (\tref{tab:histone-gene-codonusage} and data not shown).
    This suggests that histone translation may be sensitive to tRNA abundance.

  \subsection{Histone variant genes}
    The \TotalCoreVariantGenes{} annotated human histone variant genes
    are listed in \tref{tab:variant-catalogue} for completeness.
    Histone variant gene families comprise only one or a few copies
    dispersed across the genome outside the canonical histone clusters.
    For example the three H3.3 variant encoding genes are located
    on chromosomes 1, 12, and 17, far removed from other histone genes.

    Most variant gene symbols have a separate 3 part nomenclature
    consisting of histone type, F for family,
    and an identifier letter (\fref{fig:nomenclature-variant}).
    Increasing interest in histone variant function \citep{MazeAllis2014}
    coupled with a variety of usages and conflicts between species
    has led to guidelines for improved consistency in histone variant nomenclature
    \citep{Talbert2012}.
