\section{Introduction}

  Histones are among the most abundant proteins in eukaryotic cells and contribute up to half the mass of chromatin. 
  They define the structure and accessibility of the nucleosome as the fundamental repeating unit of genome organisation 
  around which the DNA is wrapped \addref{Luger1997 Nature}.
  In addition, the many chemically reactive sidechains of histones are post-translationally modified 
  as a nexus for signalling and heritable epigenetics. 

  Relationships within the histone family have been described using a variety of terminologies
  reflecting various biochemical, functional and genomic distinctions summarised below (\tref{tab:histone-divisions}).
  
  From a biochemical perspective, histones are abundant proteins 
  that are readily isolated due to their highly basic chemical character. 
  Successive improvements in fractionation ultimately revealed 5 main histone types 
  with nomenclature H1, H2A, H2B, H3 and H4 \citep{nomenclature}.
  An additional H1--related histone H5 is recognised in avian erythrocytes \citep{HFive-review}.

  The demonstration of the nucleosome as the fundamental repeating unit of chromatin \addref{Kornberg} 
  revealed that H2A, H2B, H3 and H4 associate as an octamer of two copies each within the
  nucleosome core particle. These four histones are referred to as ``core histones''. 
  In contrast, H1 and relted H5 associate with the linker DNA between nucleosome core particles 
  and are referred to as ``linker histones''. 
  The 5 human somatic H1 isoforms and 6 tissue-specific variants have been described and catalogued in \addref{HarshmanFreitas2013}.

  Arginine and lysine content was used as an early distinction between the histones. 
  The H1 linker histone has a low arginine/lysine ratio of \LinkerArgLysRatio{} and became known as ``lysine-rich'' 
  whereas the remaining 4 core histones are ``arginine-rich'' 
  with highest ratios of \HTwoAArgLysRatio{} in H2A,  \HTwoBArgLysRatio{} in H2B, 
  \HThreeArgLysRatio{} in H3 and \HFourArgLysRatio{} in H4 type isoforms.

  Extracting histones and separating them by polyacrylamide gel electrophoresis 
  using the strongly anionic detergent sodium dodecyl sulphate (SDS PAGE) and neutral buffers 
  gives single bands for each histone type. 
  However, PAGE with non-ionic detergent Triton X--100 and uncharged urea as denaturants
  in acetic acid buffer (TAU or AUT PAGE) allows the separation of histone types into multiple bands 
  due to post-translational modifications and primary sequence variations \citep{PAGEND}. 
  These separations are reflected in subtype designations including H2A.1, H2A.2, H3.1, H3.2 and H3.3. 
  Some TAU PAGE bands still encompass considerable protein isoform heterogeneity \addref{}.

  From a functional perspective, two classes of histone can be delineated within the core histones (\tref{tab:typical-histone-differences}). 
  ``Canonical'' histones contribute to the bulk structure and generic function of chromatin. 
  Their expression is significantly elevated during S~phase to provide chromatin packaging 
  for DNA duplicated during replication. This led to their description as ``replication dependent'', 
  even though a supply of canonical histones is required throughout the cell cycle. 
  Human and other metazoan canonical histone genes are distinctive 
  because they lack introns and give rise to non-polyadenylated proten coding transcripts 
  where turnover is independently regulated via a highly conserved 3' stem-loop
  
  In contrast, ``variant'' histones such as H2A.Z, TH2B and H3.3 have less sequence identity and lower abundance. 
  They play functionally specific roles and are mostly expressed outside S~phase, 
  so are labelled as ``replication independent''. 
  Since histone variants are interpreted as taking the place of equivalent canonical histone types, 
  the variant histones are also referred to as ``replacement'' histones.
  
  From a genomic perspective the canonical histone genes are found in four clusters, possibly for regulatory efficiency. 
  The multiple gene copies in the clustered arrays are sometimes confusingly referred to as ``allellic'' 
  and the resulting protein isoforms are often referred to as ``wild type'' 
  although both genes and protein products encompass distinctive variation in sequence and relative abundance, 
  and their functional equivalence has not been tested.
  In contrast, almost all variant histones are encoded by single genes dispersed in the genome 
  with typical properties including introns, alternative splicing and polyadenylated transcripts.
  
  Despite the importance of histones for chromatin organisation and extensive interest
  in their role in epigenetics and regulation, the systematic curation and classification of human histone
  gene and protein sequences has not been revisited for some time \citep{Marzluff02}.
  In the last decade the draft human genome has been finalised 
  and rich annotations continue to propagate into reference gene and protein sequence databases (\tref{tab:difference-from-Marzluff02}).

  In this manuscript we provide a comprehensive catalogue of the nomenclature and distribution 
  of canonical histone genes, proteins and pseudogenes. 
  Since curation and annotation are dynamic and evolving, we have implemented an accompanying resource 
  that can update itself based on the most current data from the NCBI RefSeq database 
  to retain its value as a reference source in an accessible format.
  All automatically generated values are displayed with a light grey background.
  This manuscript generation has remained stable in our laboratory for several years
  and represents an example of ``reproducible research'' \addref{SchwabKarrenbachClaerbout2000?}
  that could be applied to other gene families.

  \begin{table*}
    \caption{Terminology describing histone variation}
    \label{tab:histone-divisions}
    \centering
    \begin{tabular}{F p{\dimexpr(\textwidth-\eqboxwidth{firstentry}-4\tabcolsep)}}
      \toprule
	  Allelic variants &
	  Copies of a canonical histone type gene. Not alleles by genetic definition. Not histone variants. 
	  \\
      \addlinespace
	  Canonical histones &
	  Core histones with properties described in \tref{tab:typical-histone-differences} contributing the majority of histones in chromatin. 
	  Encompasses multiple isoforms.
	  \\
      \addlinespace
	  Heteromorphous variants &
	  Histone variants with distinct function and localisation that are readily separated (e.g. SDS PAGE).
	  \\
      \addlinespace
	  Homomorphous variants &
	  Canonical histone subtypes with isoform variation requiring high resolution separation methods (e.g. TAU PAGE).
	  \\
      \addlinespace
	  Families &
	  Histone types.
	  \\
      \addlinespace
	  Isoforms &
	  Proteins with high sequence identity and equivalent function. Functional equivalece has not been demonstrated for canonical histone isoforms.
	  \\
      \addlinespace
	  Non-allelic variants &
	  Histone variants. \\
      \addlinespace
      Replacement histones &
	  Histone variants. \\
      \addlinespace
	  Replication-dependent histones &
	  Canonical histones. Expressed primarily during S phase. \\
      \addlinespace
	  Replication-independent histones &
	  Histone variants. Expressed at specific times during cell cycle outside S phase. \\
      \addlinespace
	  Subtypes &
	  Canonical histone type isoforms separable by TAU PAGE. 
	  \\
      \addlinespace
	  Types &
	  Histone sequences able to participate in specific combinations that define nucleosome structure. Related in protein sequence.
	  The 5 histone types are H1, H2A, H2B, H3 and H4.
	  \\
      \addlinespace
	  Variants &
	  Core histones with properties described in \tref{tab:typical-histone-differences} 
	  performing specialised functions and contributing minor proportions of histones in chromatin. \\
      \addlinespace
	  Wild type histones &
	  Canonical histones. \\
      \bottomrule
    \end{tabular}
  \end{table*}

  \begin{table*}
    \caption{General properties of canonical and variant histone proteins.}
    \label{tab:typical-histone-differences}
    \centering
    \begin{tabular}{l l l}
      \toprule
      \null                     & Canonical             & Variants \\
      \midrule
      Expression timing         & Replication dependent & Replication independent \\
      Sequence identity         & High                  & Low \\
      Functional relationships  & Isoforms              & Distinct specialised functions \\
      Transcript stabilisation  & Stem--loop in 3' UTR  & poly--A tail \\
      Gene distribution         & Clusters              & Scattered \\
      \bottomrule
    \end{tabular}
  \end{table*}

