\section{Introduction}
  %% Flow of introduction: histones are big part of the chromatin, the building unit.
  %% Then first division, split by type. When we know the type, we can explain
  %% nucleosome structure, which leads to core vs linker division. Then more history,
  %% we found variants. New division of classes, canonical vs variant. Explain the
  %% mess of variants, with all the names used for each division, and sometimes
  %% clashing nomenclature. Finalize with what this paper is about.

  Histones are among the most abundant proteins in eukaryotic cells and contribute up
  to half the mass of chromatin. They define the structure and accessibility of the
  nucleosome as the fundamental repeating unit of genome organisation around which
  the DNA is wrapped upon. In addition, the reactive
  side chains of histones are post-translationally modified as a nexus for signalling
  and potentially for heritable epigenetics.

  Histones have been studied for many decades as abundant proteins which were readily
  isolated due to their highly basic chemical character. Successive improvements in
  fractionation revealed 5 main histone types under a nomenclature as H1, H2A, H2B, H3
  and H4 \citep{nomenclature}. An additional histone H5 related to H1 is found in
  nucleus--containing avian erythrocytes and has been of research interest due to its
  elevated binding stability \citep{HFive-review}.
  Since the ratio of arginine/lysine is \LinkerArgLysRatio{}
  for H1, these became known as ``lysine-rich'' whereas the remaining 4 histones are
  ``arginine-rich'' with the highest ratios for core tetramer of H3--H4 with \HThreeArgLysRatio{}
  and \HFourArgLysRatio{}, and lower for the outer H2A--H2B dimers with \HTwoAArgLysRatio{}
  and \HTwoBArgLysRatio{}.

  The demonstration of the nucleosome as the fundamental repeating unit of chromatin revealed
  that H2A, H2B, H3 and H4 associate as an octamer of two copies each within the
  nucleosome core particle, so they are referred to as ``core histones''. H1 and H5
  associate with the linker DNA between nucleosome core particles and are referred to
  as ``linker histones''.

  Extracting histones from cells and separating them by polyacrylamide gel electrophoresis
  (PAGE) using the strongly anionic detergent sodium dodecyl sulphate (SDS) and neutral
  buffers gives single bands for each histone type. However, PAGE with acetic acid in gels
  containing the non-ionic detergent Triton X--100 and uncharged urea as denaturants
  (TAU or AUT) allows the separation of histones into multiple species due to both
  post-translational modifications and variations in sequence within types \citep{PAGEND}.

  There are two recognised classes of histone protein sequence variation as summarised
  in \tref{tab:typical-histone-differences}. ``Canonical'' histones contribute to the
  bulk structure and generic function of the nucleosome. They are mainly expressed
  during S~phase to provide chromatin packaging after after DNA replication,
  and are referred as ``replication dependent''. In contrast, ``variant'' histones
  have lower aggregate abundance and show less identity in alignments. They play
  functionally distinctive roles, are mostly expressed outside S~phase of the cell
  cycle and are also referred as ``replacement'' or ``replication independent'' histones.

  \begin{table*}
    \caption{General properties of canonical and variant histone proteins.}
    \label{tab:typical-histone-differences}
    \centering
    \begin{tabular}{l l l}
      \toprule
      \null                     & Canonical             & Variants \\
      \midrule
      Expression timing         & Replication dependent & Replication independent \\
      Sequence identity         & High                  & Low \\
      Functional relationships  & Isoforms              & Distinct specialised functions \\
      Transcript stabilisation  & Stem--loop in 3' UTR  & poly--A tail \\
      Gene distribution         & Clusters              & Scattered \\
      \bottomrule
    \end{tabular}
  \end{table*}

  The term ``variant histones'' has come to imply canonical histones are homogeneous
  and masks their heterogeneity. Each variant, such as H2A.Z, H2Abbd, and H3.3,
  is encoded by a gene, with its own sequence and name,
  whereas each of the canonical histones are encoded by multiple genes
  with less varying sequence.
  Because their resulting protein isoforms differ at only a few sites, they are often
  referred to as ``wild type'' sequences.
  When their multiple isoforms are not treated as a single protein, they are divided
  into``sub-types'' such as H2A.1 and H2A.2 but this division has no functional or sequence
  interpretation. To increase the confusion, sub-types such as H2A.2, are also sometimes incorrectly
  referred to as ``variants''.

  The genes encoding canonical histone isoforms are typically found in clusters in most eukaryotic
  genomes, possibly for regulatory efficiency and to facilitate their sequence conservation. In
  contrast, most variant histones are encoded as single genes outside of the histone clusters.
  This distinction led to yet another distinct name, allelic and non-allelic variants for
  canonical isoforms and variant histones, respectively.

  %% on the choice of words of "protein coding transcripts":
  %%    things such a non-coding RNA are also transcripts and don't have a poly-A
  %%    tail. We could say that there are few transcripts without poly-A tail but
  %%    that would leave people wondering what are the other other proteins whose
  %%    transcripts do not have it.
  The canonical and variant classes of histones are also distinguished by their regulation in humans and other
  metazoans. Like other protein coding transcripts, histone variants
  are 3' polyadenylated. On the other hand, canonical histones are the only transcripts
  known to be non polyadenylated, and instead terminate in a highly conserved stem-loop that
  provides them with an independent basis for regulation of turnover.

  This alternative unique system seems to be limited to metazoans, with exceptions
  in the multicellular differentiated alga \textit{Volvox carteri} \citep{volvox-h2a-h2b-report}
  and the unicellular \textit{Chlamydomonas reinhardtii} \citep{chlamydomonas-histones}.
  Both are green algae from the order Chlamydomonadales, which is placed at an early
  ancestor of the higher plants.
  Other organisms such as ciliate, fungi, and higher plants, exhibit typical
  polyadenylated canonical histones transcripts.

  %% About Volvox carteri (from volvox-h2a-h2b-report)
  %% Although rRNA sequence comparisons suggest that
  %% V. carteri is phylogenetically more closely linked to higher
  %% plants than to animals (Rausch et al., 1989), the ontogenesis
  %% of this simple green alga exhibits remarkable analogies to
  %% early stages of animal embryogenesis, including a gastrulation-like
  %% inversion process (reviewed by Kirk and Harper,
  %% 1986; Kirk, 1988).

  In metazoans, the stem-loop mechanism extends to the H1 histone with an exception
  in the nematodes \textit{Caenorhabditis briggsae} and \textit{Caenorhabditis elegans}.
  However, the H1 histone in these organism already displays other diverging
  features; they are not located in histone clusters and have introns\citep{caenorhabditis-stem-loop}.

  As these are only different levels of variance, the terms used to refer to
  different groups are no more than synonymous of each other. Because of this,
  they can be easily misused without a deeper knowledge of all histone variance,
  although the correct meaning can be usually infered from the context. We
  have compiled the different terms in \tref{tab:histone-divisions}

  %% papers where H2A.2/1 are referred as variants if we want references where the
  %% the mistake is made. But really, should we be pointing the finger to them?
  %%
  %% * Galasinski, Scott C., et al. "Global regulation of post-translational
  %%   modifications on core histones." Journal of Biological Chemistry 277.4 (2002): 2579-2588.
  %% * Kimura, Hiroshi. "Histone dynamics in living cells revealed by
  %%   photobleaching." DNA repair 4.8 (2005): 939-950.
  %% * Ishibashi, Toyotaka, et al. "Acetylation of vertebrate H2A. Z and its
  %%   effect on the structure of the nucleosome." Biochemistry 48.22 (2009): 5007-5017.
  %% * Alatzas, Anastasios, and Athina Foundouli. "Alterations in core histone
  %%    variant ratios during maize root differentiation, callus formation and in
  %%    response to plant hormone treatment." Biol Res 42 (2009): 445-460.

  \begin{table*}
    %% guess this could also be done in a description environment but...
    \caption{Typical sub-divisions of histones.}
    \label{tab:histone-divisions}
    \centering
    \begin{tabular}{F p{\dimexpr(\textwidth-\eqboxwidth{firstentry}-4\tabcolsep)}}
      \toprule
      Allelic variant & Same as canonical isoform. \\
      Canonical &   All of the histones listed in \tref{tab:histone-catalogue}.
                    Their main distinct features is a peak of expression during
                    S--phase and are usually regarded as without special
                    features. \\
      \addlinespace
      Heteromorphous & Same as variant. \\
      \addlinespace
      Homomorphous & Same as sub-type. \\
      \addlinespace
      Isoform   &   Often used incorrectly to mean a variant, it should be
                    used to mean an alternate form of the same histone type
                    either within the canonical or the same variant. \\
      \addlinespace
      Non-allelic variant & Same as variant. \\
      \addlinespace
      Replacement & Same as variant. \\
      \addlinespace
      Replication-dependent & Same as canonical. \\
      \addlinespace
      Replication-independent & Same as variant. \\
      \addlinespace
      Sub-type  &   H2A and H3 histones are sometimes further divided in the
                    subtypes H2A.1, H2A.2, H3.1, and H3.2. This division should
                    be avoided since it has no interpretation. Note that H3.3
                    is an histone variant, not a sub-type. \\
      \addlinespace
      Type      &   One of the 5 histone types, H1, H2A, H2B, H3, or H4.\\
      \addlinespace
      Variant   &   All of the histones listed in \tref{tab:variant-catalogue},
                    any histone that is not a canonical. Their main distinct
                    features is an expression independent of the cell cycle as
                    opposed to canonical histones. Sometimes, also incorrectly
                    named histone isoforms. \\
      \addlinespace
      Wild type &   Usually meaning the same as canonical or sub-type 1. This
                    naming is just wrong, since it suggests that variants are
                    not natural. \\
      \bottomrule
    \end{tabular}
  \end{table*}

  Despite the importance of histones to the organisation of chromatin, and extensive interest
  in their role in epigenetics and regulation, the curation and classification of human histone
  gene and protein sequences has not been revisited for a decade \citep{Marzluff02}. In this time
  the draft human genome has been finalised and rich annotations continue to propagate through
  gene and protein sequence databases (\tref{tab:difference-from-Marzluff02}).

  Any equivalent normal approach will also become progressively more outdated from the
  time of writing. Instead, we have written this article under the concept of a live
  object, one that can be reconstructed at any time using the most up to date information
  available online.

  We have also scanned the popular RefSeq
  database to curate and report a number of oversights in untranslated regions, transcripts and
  classification of pseudo genes. Indeed, this article has been constantly
  rebuilt since 2010 as we, and others, contributed to the curation of histone
  genes.  To show the extent of this process,
  automatically generated values are displayed with a light grey background.

  A discussion about the function of each histones variants will not be presented
  as these are periodically published.

  In this paper we provide a comprehensive overview of the nomenclature and distribution
  of canonical and variant histone genes. We do not present a discussion on the
  function of the multiple variants as excellent reviews for each of them are
  periodically published. Furthermore, since curation and annotation are
  dynamic processes, we have implemented an accompanying resource that automatically generates
  the tables and figures presented in the submitted manuscript based
  on the most current data from the NCBI RefSeq database.

