\section{Materials and methods}
\label{sec:matmethods}

  The primary manuscript is generated from a \LaTeX{} source
  deriving dynamic data from automatic builds using SCons \citep{SCons2005}.
  Core processing is performed by bp\_genbank\_ref\_extractor
  which is available in the Bio-EUtilities module, part of the
  BioPerl project \citep{bioperl}.
  Processing was done using BioPerl and Bio-EUtilities versions
  \BioPerlVersion{} and \BioEUtilitiesVersion{} respectivelly.
  Data was obtained from NCBI RefSeq on \SequencesDate{} \citep{PruittRefseq2014}.
  Alignments were generated by T-Coffee version
  \TCoffeVersion{} \citep{tcoffee2000},
  sequence logos were generated using WebLogo version
  \todo{weblogo version} \citep{weblogo},
  and codons were analysed using SNAP \todo{SNAP version} \citep{KorberSNAP2000}
  and CUSP \todo{EMBOSS version} \citep{Emboss2000}.
  Sequences are represented following Human Genome Variation
  Society (HGVS) recommended nomenclature \citep{mutnomenclature2003}.
  All sources are freely available in a github repository
  \url{https://github.com/af-lab/histone-catalog}, including all
  source for figures, manuscript, and build system making public
  all parameters used in the above programs.
