\section{Materials and methods}
\label{sec:matmethods}

  The primary manuscript is generated from a LaTex source
  deriving dynamic data from automatic builds using SCons \todo{SCons version} \citep{SCons2005}.
  Core processing is performed by bp\_genbank\_ref\_extractor
  available in the Bio-EUtilities \todo{BioPerl version} \citep{BioPerl2002}.
  Data is obtained from NCBI RefSeq release \todo{RefSeq release} \citep{PruittRefseq2014}.
  Alignments were generated by T-Coffee \todo{T-Coffee version} \todo{parameters?} \citep{tcoffee2000},
  sequence logos were generated using WebLogo \todo{weblogo version} \todo{parameters?} \citep{weblogo},
  and codons were analysed using SNAP \todo{SNAP version} \citep{KorberSNAP2000}
  and CUSP \todo{EMBOSS version} \citep{Emboss2000}.
  Sequences are represented following Human Genome Variation Society recommended nomenclature \citep{mutnomenclature2003}.
  All scripts, latex sources and current sequence files
  are available in a github repository \url{https://github.com/af-lab/histone-catalog}
