\section{Histone proteins}

	It is widely implied that canonical histone proteins of each type behave as a single protein, 
	sometimes referred to as ''wild type'' \tref{tab:histone-divisions}. 
	This assumption may result from the relatively high identity of histone sequences between species 
	and the focus on functional distinctions supplied by histone variants. 
	However, the implications of strong conservation of amino acid sequence \addref{NeiRooney AnnRevGen 2005}
	and the roles of specific amino acid sequence differences \addref{MazeAllis NRG 2014} 
	is that even minor variations in canonical histone proteins could have functional implications.
	
	The sequence variation in isoforms of the canonincal histone proteins H2A, H2B H3 and H4 
	is shown in figures \todo{reference protein alignment figures}.
	Isoform variation in canonical histone types H2A and H2B is most pronounced, 
	with \HTwoAUniqueProteins{} H2A and \HTwoBUniqueProteins{} H2B proteins respectively. 
	For H3 and H4 there are \HThreeUniqueProteins{} and \HFourUniqueProteins{} encoded. 
	Typically, human canonical histone polypeptides are numbered omitting the N-terminal methionine 
	becuase this is likely to be removed since most sequences have a small hydrophilic amino acid as the second residue.
	\todo{We should change ths script here because it will confuse people that H3K4 is at position 5 ...}
    %% if we want to change our stance in this, we only need to uncomment
    %% 3 lines in MyLib.pm, in the routine that reads the sequence files.

  \subsection{H2A isoforms}
	
	The canonical H2A protein isoforms are separable on TAU PAGE into two bands labelled as H2A.1 and H2A.2. 
	This separation in fact distingushes H2A with residue 52 as leucine or methionine \addref{???}, 
	so there are actually 8 different protein sequences in H2A.1 and 2 different sequences in H2A.2. 
	There is no concordance between genes encoding H2A.1 and H2A.2 and their location histone gene clusters. 
	HIST2 contains genes encoding isoforms with both L52 and M52, 
	and clusters HIST1, HIST2 and HIST3 all encode canonical H2A isoforms with L52. 
	There is also no known functional distinction between H2A.1 and H2A.2, 
	so this histone subtype nomenclature based on an arbitrary TAU PAGE separation is misleading.

	As demonstrated in \ref{fig:H2A-weblogo}, the major sites of difference in canonical H2A are 
	serine or threonine at residue 17, 
	alanine or serine at residue 41, 
	lysine, arginine or glycine at residue 100, 
	and variation in the C-terminal residues from residue 125 including terminal deletions. 
	In principle all of these residue are have implications for post-translational modifications.
	
  \subsection{H2B isoforms}

	H2B is sometimes referred to as being less variable in sequence than H2A and H4. 
	Although there are very few H2B histone variants, this histone has the most canonical isoforms (\tref{tab:histone-divisions}).

	As discussed above, the \textit{HIST1H2AA} and \textit{HIST1H2BA} genes 
	are the most divergent of all canonical isoforms.
	They are located together close to, but not strictly within, the HIST1 cluster. 
	HIST1H2BA is known to be testes specific TSH2B and is structurally distinctive \addref{Uruhara ACTAcrystB 2014}. 
	It has been proposed to label this histone isoform as H2B.1 \addref{Talbert EpigeChrom 2012}. 
	although it is unclear whether this isoform separable by TAU PAGE like the other subtype nomenclatures.

	In fact, none of the canonical H2B appear to be separable by TAU PAGE 
	despite this band encompassing up to \HTwoBUniqueProteins{} unique isoforms. 
	There is significant variability between isoforms in the N-terminus, 
	especially at residue 3 which can be aspartic or glutamic acid, 
	and at residue 5 which can be alanine, serine, threonine or valine depending on the isoform. 
	This region is also one of the most variable between canonical core histones of different species. 
	%% I fudged the preceding sentence to avoid having to discuss the PEP thing, since we are only concenrned with human histones
	Although the physicochemical distinction between valine or isoleucine at residue 40 is small, 
	isoform differences of glycine or serine at residue 76 
	and serine or alanine at 125 could again impact on post-translational modification pootential.
	
  \subsection{H3 isoforms}
	
	Although H3 histone variants are of considerable interest (\tref{tab:histone-variant-families}), 
	there are only \HThreeUniqueProteins{} different canonical H3 isoforms. 
	The majority of H3 genes are in the HIST1 cluster 
	and encode a single polypeptide sequence. 
	In contrast, the three canonical H3 genes in HIST2 encode an isoform 
	with the sole but interesting difference of serine rather that cysteine at residue 97.
	These two isoforms can also be separated by TAU PAGE based on residue 97. 
	Therefore, H3.2 isoforms are unique and are encoded in the HIST2 cluster, 
	whereas H3.1 isoforms are likely to be almost entirely unique and encoded in the HIST1. 
	The sole complication is that HIST3 encodes a single canonical H3 isoform 
	with four amino acid differences.
	
  \subsection{H4 isoforms}
	H4 is the most homegenous of all canonical core histones, 
	with all but one of the \HFourCodingGenes{} genes encoding identical proteins. 
	These genes are located across HIST1, HIST2 
	and the isolated canonical H4 gene known as HIST4.
	However, the \textit{HIST1H4G} gene in the middle of the HIST1 cluster 
	encodes a significantly divergent H4 isoform 
	with lo less than 15 amino acid differences and a deletion of the C-terminal 5 residues. 
	This gene is annotated as being transcribed \todo{which tissues?}
	so could be of considerable interest for further investigation 
	since no H4 histone variants have been recognised.

\todo{The table of differences and logo need to be combined as parts A and B of single figures for each histone type}
\todo{Alignments are not handling insertions in an ideal way? We need to omit single insertions eg HIST1H2BA.}
\todo{There is an extra comma immedaitely after the HISTnHx in every table legend that is confusing and needs to be removed}

  \begin{TableAndFigure*}
    \captionof{table}{Description of all H2A isoforms relative to the most
                      common protein sequence. Description of isoforms
                      follows the latest recommendations from the Human
                      Genome Variation Society \citep{desc-seq-variant}.}
    \label{tab:H2A-consensus}
    \input{results/table-H2A-proteins-align}

    \includegraphics[width=\textwidth]{seqlogo_H2A_proteins.pdf}
    \captionof{figure}{Probability sequence logo with all H2A sequences. The
                       sequence logo was generated using \citep{weblogo} after
                       alignment with T--Coffee \citep{tcoffee2000}. To ease
                       the identification of variantion, locations where
                       none occurs was greyd out.}
    \label{fig:H2A-weblogo}
  \end{TableAndFigure*}

  \begin{TableAndFigure*}
    \captionof{table}{Description of all H2B isoforms relative to the most
                      common protein sequence. See caption of
                      \tref{tab:H2A-consensus} for details.}
    \label{tab:H2B-consensus}
    \input{results/table-H2B-proteins-align}

    \includegraphics[width=\textwidth]{seqlogo_H2B_proteins.pdf}
    \captionof{figure}{Probability sequence logo with all H2B sequences.
                       See the caption of \tref{fig:H2A-weblogo} for details.}
    \label{fig:H2B-weblogo}
  \end{TableAndFigure*}



  \begin{TableAndFigure*}
    \captionof{table}{Description of all H3 isoforms relative to the most
                      common protein sequence. See caption of
                      \tref{tab:H2A-consensus} for details.}
    \label{tab:H3-consensus}
    \input{results/table-H3-proteins-align}

    \includegraphics[width=\textwidth]{seqlogo_H3_proteins.pdf}
    \captionof{figure}{Probability sequence logo with all H3 sequences.
                       See the caption of \tref{fig:H2A-weblogo} for details.}
    \label{fig:H3-weblogo}
  \end{TableAndFigure*}

  \begin{TableAndFigure*}
    \captionof{table}{Description of all H4 isoforms relative to the most
                      common protein sequence. See caption of
                      \tref{tab:H2A-consensus} for details.}
    \label{tab:H4-consensus}
    \input{results/table-H4-proteins-align}

    \includegraphics[width=\textwidth]{seqlogo_H4_proteins.pdf}
    \captionof{figure}{Probability sequence logo with all H4 sequences.
                       See the caption of \tref{fig:H2A-weblogo} for details.}
    \label{fig:H4-weblogo}
  \end{TableAndFigure*}
