\section{Histone proteins}

	Most depictions of chromatin imply that 
	canonical histone protein types behave as a single protein, 
	sometimes referred to as ``wild type''. 
	The assumption arises from limited awareness of isoform variation, 
	the relatively high identity of histone sequences between isoforms and species, 
	and the focus on functional distinctions of histone variants. 
	However, the strong selection pressure on histone amino acid sequences \citep{birth-death-review},
	the roles of specific amino acid differences \citep{MazeAllis2014}, 
	and the structural effects of small variations \citep{KurumizakaCOSB2013} 
	suggests that even minor variations in canonical histone proteins 
	could have functional implications.
	
	Isoform variation in canonical histone types H2A and H2B is most pronounced, 
	with potentially \HTwoAUniqueProteins{} different H2A 
	and \HTwoBUniqueProteins{} different H2B proteins respectively. 
	In contrast only \HThreeUniqueProteins{} H3 and \HFourUniqueProteins{} H4 proteins 
	are encoded for these canonical histone types. 
	There is also some variation in canonical histone length, particularly for H2A and H2B isoforms.

	The nomenclature for histone protein isoforms follows directly from the gene names \citep{Marzluff02}, 
	and supersedes the earlier nomenclature with backslash which assigned different isoform letters \citep{AlbigGenomics1997,AlbigHumangen1997}.
	Human canonical histone polypeptides are usually numbered 
	with ommision of the N-terminal methionine 
	which is likely to be removed since most sequences have 
	a small hydrophilic amino acid as the following residue \citep{XiaoPeiBiochem2010} 
	and has been ommitted here for consistency.
	\todo{Please change the script here}

  \subsection{Canonical H2A isoforms}
	
	Canonical H2A genes encode \HTwoAUniqueProteins{} different H2A isoforms \ref{fig:H2Aisoforms}.
	This set of proteins is separable by TAU PAGE into two bands identified as H2A.1 and H2A.2. 
	The TAU PAGE separation distinguishes leucine or methionine at residue 52 \citep{FranklinZweidler1977,Zweidler1977}, 
	implying there are up to 8 different protein sequences in H2A.1 and 2 in H2A.2. 
	There is no concordance between H2A.1 and H2A.2 
	and the location of their isoform-encoding genes in the histone gene clusters. 
	HIST2 contains genes encoding isoforms with both Leu52 and Met52, 
	and HIST1, HIST2 and HIST3 all contain genes encoding H2A isoforms with Leu52. 
	No clear functional distinction between H2A.1 and H2A.2 has yet been reported.

	Excluding the variations of \textit{HIST1H2AA} discussed above, 
	the major sites of difference in canonical H2A are 
	serine or threonine at residue 17, 
	alanine or serine at residue 41, 
	lysine or arginine at residue 100, 
	and the C-terminal residues from residue 125 onwards \ref{fig:H2Aisoforms}. 
	All these sites have implications for post-translational modifications. 
	
  \begin{Figure*}
	  \label{tab:H2Aisoforms}
	\caption{Canonical H2A protein isoforms.
	 Upper panel shows relative to the most common protein sequence using standard mutation nomenclature \citep{mutnomenclature2003}. 
	 Lower panel shows sequence logo from isoform alignment \citep{weblogo,tcoffee2000} with invariant residues in grey.}
	 \input{results/table-H2A-proteins-align}
	 \includegraphics[width=\textwidth]{seqlogo_H2A_proteins.pdf}
  \end{TableAndFigure*}

  \subsection{Canonical H2B isoforms}

	Canonical H2B has the most canonical isoforms \ref{fig:H2Aisoforms} 
	although none of these \HTwoBUniqueProteins{} unique proteins appear to be separable by TAU PAGE.

	There is significant variability between isoforms in the N-terminal region. 
	This is one of the most variable between canonical core histones of different species, and
	human H2B has a unique and distinctive N-terminal proline-acidic-proline motif. 
	The residue following this PEP/PDP motif is very variable with either alanine, serine, threonine or valine. 

	Excluding the variations of \textit{HIST1H2BA} discussed above, 
	the isoform differences are mainly the chemically similar valine or isoleucine at residue 40, 
	and serine to glycine or alanine at residues 76 and 125 which have post-translational modification implications. 
	Variation in H2B isoform levels between multiple cancer cell lines has been observed \citep{Molden2015}, 
	although the functional implications are unknown.

  \begin{Figure*}
	  \label{tab:H2Bisoforms}
	\caption{Canonical H2B protein isoforms.
	 Upper panel shows relative to the most common protein sequence using standard mutation nomenclature \citep{mutnomenclature2003}. 
	 Lower panel shows sequence logo from isoform alignment \citep{weblogo,tcoffee2000} with invariant residues in grey.}
	 \input{results/table-H2B-proteins-align}
	 \includegraphics[width=\textwidth]{seqlogo_H2B_proteins.pdf}
  \end{TableAndFigure*}

  \subsection{Canonical H3 isoforms}
	
	Canonical H3 genes encode only \HThreeUniqueProteins{} different canonical isoforms \ref{fig:H2Aisoforms}. 
	The majority of H3 genes are in the HIST1 cluster and encode a single polypeptide sequence \citep{Ederveen2011}. 
	The three canonical H3 genes in HIST2 encode a different isoform 
	with the interesting difference of serine rather that cysteine at residue 97 
	that is separable by TAU PAGE \citep{FranklinZweidler1977}.
	By coincidence the HIST1-encoded canonical H3 isoforms 
	are labelled as H3.1 and the HIST2-encoded copies are labelled H3.2.
	As discussed above, HIST3 appears to be a largely testes specific variant
	and encodes a single canonical H3 isoform with four amino acid differences. 
	
  \begin{Figure*}
	\label{tab:H3isoforms}
	\caption{Canonical H3 protein isoforms.
	 Upper panel shows relative to the most common protein sequence using standard mutation nomenclature \citep{mutnomenclature2003}. 
	 Lower panel shows sequence logo from isoform alignment \citep{weblogo,tcoffee2000} with invariant residues in grey.}
	 \input{results/table-H3-proteins-align}
	 \includegraphics[width=\textwidth]{seqlogo_H3_proteins.pdf}
  \end{TableAndFigure*}

  \subsection{Canonical H4 isoforms}
	H4 is the most homegenous of all canonical core histones, 
	with all but one of the \HFourCodingGenes{} genes encoding identical proteins. 
	These genes are located across HIST1, HIST2 
	and the isolated canonical H4 gene known as HIST4.
	The divergent \textit{HIST1H4G} gene in the middle of the HIST1 cluster 
	encodes an isoform with 15 amino acid differences and a deletion of the C-terminal 5 residues. 
	This gene is annotated as transcribed so merits further investigation.

  \begin{Figure*}
	\label{tab:H4isoforms}
	\caption{Canonical H4 protein isoforms.
	 Upper panel shows relative to the most common protein sequence using standard mutation nomenclature \citep{mutnomenclature2003}. 
	 Lower panel shows sequence logo from isoform alignment \citep{weblogo,tcoffee2000} with invariant residues in grey.}
	 \input{results/table-H4-proteins-align}
	 \includegraphics[width=\textwidth]{seqlogo_H4_proteins.pdf}
  \end{TableAndFigure*}

\todo{There is an extra comma immedaitely after the HISTnHx in every table legend that is confusing and needs to be removed}
