\section{The proteins}

  The most common misconception about canonical histones is that canonical
  histones are only 4 proteins. As previosly discussed, this masks the true
  diversity of histone, and while some genes do encode the same protein, the
  opposite is far more common with \HTwoAUniqueProteins{} H2A,
  \HTwoBUniqueProteins{} H2B, \HThreeUniqueProteins{} H3, and
  \HFourUniqueProteins{} H4 unique protein sequences.

  Studies that only make reference to the histone type,
  are incomplete on their description. To make matters worse, the existence
  of sub-types, such as H2A.1 and H2A.2, is even more misleading since those
  who might be aware of canonical variance may mistake them for true
  representatives of it while in truth they are of no meaning.

  We have collected all the proteins for all canonical core histones, aligned
  them, and created a sequence logos for each of them (\fref{fig:H2A-weblogo},
  \ref{fig:H2B-weblogo}, \ref{fig:H3-weblogo}, and \ref{fig:H4-weblogo}) which
  gives a compact overview where variation within closely related multiple
  sequences occur. In addition, we also have described each of the isoforms
  using as reference the most common sequence (\tref{tab:H2A-consensus},
  \ref{tab:H3-consensus}, \ref{tab:H3-consensus}, and \ref{tab:H4-consensus}).
  As with all values in this paper, both the tables and figures are generated
  automatically from the sequences available online as reference.


  The logica is that variance between canonical isoforms is too small to have a likely
  different function. However, the variable H3.3 is only slightly different from the
  canonical H3 and seems to be special.


  It is an attractive idea that H3 and H4, which are at the core of the core,
  have less variance on their sequence as result of optimized sequence for stability
  while H2A, and H2B don't have to suffer such constraint.

  Most of the PTM are in the histone tails but a few are within the histone fold (H3 T45 for example)

  \subsection{Sequence}
    When dealing with sequence of histone proteins, it is standard practice to
    ignore the start methionine as this is likely to be cleaved. The end
    result is that typical reference to specific amino acids, such as H2AX~S139
    or H3~K4, are actually referent to the amino acids at position 140 and 5.

    %% if we want to change our stance in this, we only need to uncomment
    %% 3 lines in MyLib.pm, in the routine that reads the sequence files
    It is a standard guideline to use the primary translation product when
    refering to protein locations, not a mature protein after
    post-translation modifications \citep{desc-seq-variant}.
    This prevents confusions, specially for anyone less
    used to this histone divergence from common standards.
    \todo{andrew said that because in H2B, the first 4 aa (MPEP) can be cleaved
          so the location is actually off by 4 but I can't see a paper with it.
          If we can find one reference of it, we could make a better case of how
          ridiculous this situation is}.
    Because of this, we are counting every single amino acid of the translated
    product, and thus depart from the ``histone specific'' counting in all
    analysis that follow.


  \subsection{Histone Fold Domain}
    \todo[inline]{Is histone protein structure too much off topic?}

    All histone proteins have a common motif named the Histone Fold Domain (HFD).
    It is composed of a long $\alpha$ helix and two loops on its side that fold
    back two other shorter $\alpha$ helices. This structure allows the histones
    to dimerize with their pair (H3--H4 and H2A--H2B) and form the histone
    octamer. The central $\alpha$ helix of each dimer pair fits in the gap between
    the $\alpha$1 and $\alpha$2 helices crossing the pair $\alpha$2 helix. In
    addition to these, H3 and H2B display a short $\alpha$N and $\alpha$C helix
    on their terminals.

    The HFD is common to all histone and shared with other
    proteins such as the CENP--TWSC and NF--Y complexes. All of these are nuclear
    proteins that interact with DNA. In the case of NF--Y, these even replace
    the H2A--H2b histones during active conditions.

    Whether these proteins actually ever happen to duplicate the nucleosome functions...

  \subsection{H2A}
    \begin{TableAndFigure*}
      \captionof{table}{histone H2A protein consensus}
      \label{tab:H2A-consensus}
      \input{results/table-H2A-align}

      \includegraphics[width=\textwidth]{seqlogo_H2A.pdf}
      \captionof{figure}{WebLogo of all H2A sequences. The bottom line are the amino acids
                         of H2AFJ whose sequence is different from any of the H2A proteins}
      \label{fig:H2A-weblogo}
    \end{TableAndFigure*}

    \todo[inline]{comment on H2AFJ}

  \subsection{H2B}
    \begin{TableAndFigure*}
      \captionof{table}{histone H2B protein consensus}
      \label{tab:H2B-consensus}
      \input{results/table-H2B-align}

      \includegraphics[width=\textwidth]{seqlogo_H2B.pdf}
      \captionof{figure}{WebLogo of all H2B sequences.}
      \label{fig:H2B-weblogo}
    \end{TableAndFigure*}

  \subsection{H3}

    All H3.1 are in the histone clusters 1 and 3. All the H3.2 are in cluster 2.
    The variant H3.3 is not in clusters.


    \begin{TableAndFigure*}
      \captionof{table}{histone H3 protein consensus}
      \label{tab:H3-consensus}
      \input{results/table-H3-align}

      \includegraphics[width=\textwidth]{seqlogo_H3.pdf}
      \captionof{figure}{WebLogo of all H3 sequences.}
      \label{fig:H3-weblogo}
    \end{TableAndFigure*}

  \subsection{H4}
    %% these should all be the same but they are not. And there's an H4 in HIST4
    %% which is also present in mouse, which means has been conserved  in evolution
    With the exception of HIST1H4G, all H4 genes seem to encode the same protein. Even HIST4H4 which is the
    only gene on that ``cluster''.
    \begin{TableAndFigure*}
      \captionof{table}{histone H4 protein consensus}
      \label{tab:H4-consensus}
      \input{results/table-H4-align}

      \includegraphics[width=\textwidth]{seqlogo_H4.pdf}
      \captionof{figure}{WebLogo of all H4 sequences.}
      \label{fig:H4-weblogo}
    \end{TableAndFigure*}

