\section{Histone proteins}
  Most depictions of chromatin imply that
  canonical core histone protein types behave as a single protein,
  often referred to as ``wild type''.
  The assumption is based on the relatively high identity of histone sequences
  between isoforms and species,
  and historical interest in functional roles of histone variants.

  However, the very strong selection pressure on
  amino acid sequences encoded by canonical core histone genes \citep{NeiRooney2005},
  the roles of specific amino acid differences in observed proteins \citep{MazeAllis2014},
  and the consequences of small variations in proteins on the structure of nucleosomes \citep{KurumizakaCOSB2013}
  all suggest that the minor differences in the encoded canonical core histone protein isoforms
  could have functional implications.

  Encoded isoform variation in canonical histone types H2A and H2B is most pronounced,
  with \HTwoAUniqueProteins{}~H2A and \HTwoBUniqueProteins{}~H2B distinct polypeptides
  for these histone type genes.
  In contrast, only \HThreeUniqueProteins{}~H3 and \HFourUniqueProteins{}~H4
  distinct polypeptides are encoded by similar numbers of genes (\tref{tab:histone-gene-count}).
  Although most variation is a result of amino acid substitutions,
  some H2A and H2B isoforms also show length differences.

  The nomenclature for histone protein isoforms follows directly
  from the gene names \citep{Marzluff02},
  and supersedes the earlier nomenclature with forward slash which used different isoform letters
  \citep{AlbigGenomics1997,AlbigHumangen1997}.
  The encoded proteins described below are products of genes and transcripts listed
  in table \tref{tab:histone-catalogue}.
  Human canonical core histone polypeptides have typically been numbered
  with omission of the N-terminal methionine
  since this is likely to be removed because most sequences have
  a small hydrophilic amino acid as the second residue \citep{XiaoPeiBiochem2010}.
  This convention predates the Human Genome Variation Society (HGVS) recommendation
  to include the initial methionine as residue 1.
  We have omitted the N-terminal methionine on figures, alignments,
  and amino-acid numbering for consistency.

%% To include methionine just uncomment 3 lines in the routine that reads the sequence files in MyLib.pm

  \subsection{Canonical H2A isoforms}
    Canonical H2A genes encode \HTwoAUniqueProteins{}~different
    protein isoforms with pairwise identity down to
    \FPround{\result}{\HTwoAPID}{0} \SI{\result}{\percent} (\tref{tab:H2A-consensus}).
    These are separable by TAU PAGE into two bands identified as H2A.1 and H2A.2.
    TAU PAGE distinguishes leucine from methionine at residue 51 \citep{FranklinZweidler1977,Zweidler1977},
    implying there are up to 8 different protein sequences in H2A.1 and 3 in H2A.2.
    There is no concordance between H2A.1 and H2A.2
    and the location of isoform-encoding genes in the histone gene clusters.
    The HIST2 cluster contains genes encoding isoforms with both Leu51 and Met51,
    while HIST1, HIST2, and HIST3 clusters all contain genes encoding H2A isoforms with Leu51.
    No functional distinction between H2A.1 and H2A.2 has been reported.

    Excluding \textit{HIST1H2AA} discussed above,
    the sites of difference in two or more isoforms of canonical H2A are
    serine or threonine at residue 16,
    alanine or serine at residue 40,
    lysine, arginine, or glycine at residue 99,
    and the C-terminal residues from 124 onwards (\tref{tab:H2A-consensus}).
    All these sites have implications for post-translational modifications.

    \begin{table}
      \caption{%
        Canonical H2A encoded protein isoforms.
        Upper panel shows isoform variations relative to most common isoform
        using HGVS recommended nomenclature \citep{mutnomenclature2003}.
        Lower panel shows sequence logo of all isoforms aligned
        with invariant residues in grey.
      }
      \label{tab:H2A-consensus}
      \input{results/table-H2A-proteins-align}
      \includegraphics[width=\textwidth]{figs/seqlogo_H2A_proteins.pdf}
    \end{table}

  \subsection{Canonical H2B isoforms}
    Canonical H2B has more isoforms than the other histone types (\tref{tab:H2B-consensus})
    with \HTwoBUniqueProteins{}~unique proteins
    diverging to \FPround{\result}{\HTwoBPID}{0}\result\% identity.
    Nevertheless, all isoforms appear to migrate together in TAU PAGE.

    There is significant variability between isoforms in the N-terminal region
    and this is one of the most variable sites between canonical
    core histones of different species.
    Human H2B has a unique and distinctive N-terminal proline-acidic-proline motif (PEP/PDP)
    followed by a very variable residue that can be alanine, serine,
    threonine, or valine (\tref{tab:H2B-consensus}).

    Excluding \textit{HIST1H2BA} discussed above,
    the remaining isoform differences in two or more isoform are mainly the chemically
    similar valine or isoleucine at residue 39,
    and serine to glycine and alanine at residues 75 and 124 respectively,
    which have have post-translational modification implications.
    A number of H2B genes are annotated to have multiple transcripts,
    although in most cases these transcripts encode identical protein isoforms.
    Variation in H2B transcript isoform levels between multiple cancer cell
    lines has been observed \citep{Molden2015},
    although the functional implications are unknown.

    \begin{table}
      \caption{%
        Canonical H2B encoded protein isoforms.
        Upper panel shows isoform variations relative to most common isoform
        using HGVS recommended nomenclature \citep{mutnomenclature2016}.
        For clarity, isoforms encoded by multiple transcripts of a single gene
        are distingushed by a numerical suffix (see \tref{tab:histone-catalogue}).
        Lower panel shows sequence logo of all isoforms aligned
        with invariant residues in grey.
      }
      \label{tab:H2B-consensus}
      \input{results/table-H2B-proteins-align}
      \includegraphics[width=\textwidth]{figs/seqlogo_H2B_proteins.pdf}
    \end{table}

  \subsection{Canonical H3 isoforms}
    Canonical H3 genes encode only \HThreeUniqueProteins{}~different
    protein isoforms (\tref{tab:H3-consensus}).
    The majority of H3 genes are in the HIST1 cluster and encode a
    single polypeptide sequence \citep{Ederveen2011}
    whereas the three canonical H3 genes in HIST2 encode a distinct isoform
    with the interesting difference of serine instead of cysteine at residue 96
    that is separable by TAU PAGE \citep{FranklinZweidler1977}.
    By apparent coincidence this means HIST1-encoded canonical H3 isoforms are
    identified as H3.1 and the HIST2-encoded copies are H3.2.
    As discussed above, \textit{HIST3H3} with four amino acid differences
    appears to be the largely testes specific variant H3T
    that has been assigned as H3.4 in variant nomenclature \citep{Talbert2012}
    even though it would be predicted to migrate with H3.1 in TAU PAGE \citep{FranklinZweidler1977}.

    \begin{table}
      \caption{%
        Canonical H3 encoded protein isoforms.
        Upper panel shows isoform variations relative to most common isoform
        using HGVS recommended nomenclature \citep{mutnomenclature2003}.
        Lower panel shows sequence logo of all isoforms aligned
        with invariant residues in grey.
      }
      \label{tab:H3-consensus}
      \input{results/table-H3-proteins-align}
      \includegraphics[width=\textwidth]{figs/seqlogo_H3_proteins.pdf}
    \end{table}

  \subsection{Canonical H4 isoforms}
    H4 is the most homogeneous of all canonical core histones,
    with all but one of the \HFourCodingGenes{}~genes encoding
    an identical protein sequence (\tref{tab:H4-consensus}).
    These genes are located across HIST1, HIST2, and the isolated
    \textit{HIST4H4} as the sole member of HIST4.

    The divergent \textit{HIST1H4G} gene in the middle of the HIST1 cluster
    encodes an isoform with 15 amino acid differences and a deletion of the C-terminal 5 residues.
    This gene is annotated as being transcribed and merits further investigation.

    \begin{table}
      \caption{%
        Canonical H4 encoded protein isoforms.
        Upper panel shows isoform variations relative to most common isoform
        using HGVS recommended nomenclature \citep{mutnomenclature2003}.
        Lower panel shows sequence logo of all isoforms aligned
        with invariant residues in grey.
      }
      \label{tab:H4-consensus}
      \input{results/table-H4-proteins-align}
      \includegraphics[width=\textwidth]{figs/seqlogo_H4_proteins.pdf}
    \end{table}
