\section*{Supplementary Data}

  \begin{supplement}
    \caption{Differences between the current human histone genes and \citet{Marzluff02}.}
    \label{tab:difference-from-Marzluff02}
    \input{results/table-reference_comparison}
  \end{supplement}

  %% This is a very long table (ctabular), that spans multiple pages.  TeX
  %% standard floats are unable to handle it.  We could use the longtable
  %% or the xtab packages (see Section 11.8 named free tabulars of the
  %% memoir class).  We may still do that later but at the moment we use
  %% capt-of which allows to enter a caption outside a float.
  \newpage
  \captionof{supplement}{%
    Catalogue of all annotated human canonical core histone gene,
    transcript and protein IDs.
  }
  \label{tab:histone-catalogue}
  \input{results/table-histone_catalogue}

  \newpage
  \begin{supplement}
    \centering
    \caption{Sequence logo for the H2A canonical coding histone genes}
    \label{fig:h2a-histone-gene-variation}
    \includegraphics[width=1.0\textwidth]{seqlogo_H2A_cds.pdf}
  \end{supplement}
  \newpage
  \begin{supplement}
    \centering
    \caption{Sequence logo for the H2B canonical coding histone genes}
    \label{fig:h2b-histone-gene-variation}
    \includegraphics[width=1.0\textwidth]{seqlogo_H2B_cds.pdf}
  \end{supplement}
  \newpage
  \begin{supplement}
    \centering
    \caption{Sequence logo for the H3 canonical coding histone genes}
    \label{fig:h3-histone-gene-variation}
    \includegraphics[width=1.0\textwidth]{seqlogo_H3_cds.pdf}
  \end{supplement}
  \newpage
  \begin{supplement}
    \centering
    \caption{Sequence logo for the H4 canonical coding histone genes}
    \label{fig:h4-histone-gene-variation}
    \includegraphics[width=1.0\textwidth]{seqlogo_H4_cds.pdf}
  \end{supplement}

  \newpage
  \begin{supplement}
    \caption{HGNC recognised histone variant family stem names, commonly used protein names
             and names for improved consistency based on proposal by \citet{Talbert2012}.
             \textit{HIST1H2BA} and \textit{HIST3H3} are included as discussed above.}
    \label{tab:histone-variant-families}
    \centering
    \begin{tabularx}{\linewidth}{l l l >{\raggedright\arraybackslash}X}
      \toprule
      Family & Protein & \citet{Talbert2012} & Notes \\
      \midrule
      H2AFB & H2A.Bbd & H2A.B & equivalent to H2A.L \\
      H2AFJ & H2A.J & H2A.J & at HIST4 locus \\
      H2AFV & H2A.Z-2 & H2A.Z.2 & - \\
      H2AFX & H2AX/H2A.X & H2A.X & - \\
      H2AFY & macroH2A/mH2A & macroH2A & - \\
      H2AFZ & H2A.Z & H2A.Z.1 & - \\
      H2BFM & ? & H2B.M & homologous to H2B.W, X-linked\\
      H2BFS & - & - & pseudogene \\
      H2BFWT & ? & H2B.W & testes specific, X-linked \\
      H2BFX & - & - & pseudogene \\
      HIST1H2BA & TH2B/TSH2B & H2B.1 & testes specific \\
      HIST3H3 & H3T & H3.4 & testes specific \\
      H3F3 & H3.3 & H3.3 & euchromatin related \\
      CENPA & CENP-A & - & centromere-specific \\
      \bottomrule
    \end{tabularx}
  \end{supplement}

  \newpage
  \begin{supplement}
    \caption{Sequence differences between canonical histone genes.
             Average counts of non-synonymous ($d_N$) and synonymous ($d_S$)
             nucleotide substitutions between all pairwise comparisons of
             canonical core histone coding regions by histone type,
             and average $d_N/d_S$ ratios for the pairwise comparisons as
             computed by PAML's \texttt{codeml} (\citet{GoldmanYang1994} method).}
    \label{tab:histone-gene-differences}
    \centering
    \begin{tabular}{l l l l}
      \toprule
      \null & \centercell{$d_N$} & \centercell{$d_S$} & \centercell{$d_N/d_S$} \\
      \midrule
      H2A & \MeanHTwoAdN  & \MeanHTwoAdS  & \MeanHTwoAdNdS \\
      H2B & \MeanHTwoBdN  & \MeanHTwoBdS  & \MeanHTwoBdNdS \\
      H3  & \MeanHThreedN & \MeanHThreedS & \MeanHThreedNdS \\
      H4  & \MeanHFourdN  & \MeanHFourdS  & \MeanHFourdNdS \\
      \bottomrule
    \end{tabular}
  \end{supplement}

  \newpage
  \newpage
  \captionof{supplement}{%
    Frequency of each codon relative to the number of codons coding for the
    same amino acid.
  }
  \label{tab:histone-gene-codonusage}
  \input{results/table-codon_usage}

  \newpage
  \captionof{supplement}{Catalogue of human variant core histone gene, transcript and protein IDs}
  \label{tab:variant-catalogue}
  \input{results/table-variant_catalogue}

  \newpage
  \begin{supplement}
    \caption{Analysis of the current histone curation.  Existing annotations
      are matched against current knowledge of histones to create a list
      of anomalies.  These are then reported back to RefSeq for review.}
    \label{tab:curation-anomalies}
    \begin{itemize}
    \input{results/histone_insanities.tex}
    \end{itemize}
  \end{supplement}
