\section{Histone transcripts}
  %% There is usually reference to histones being the only protein coding
  %% genes transcribed by Pol II that lack poly(A).  However, only Pol II
  %% transcribe protein coding genes so there's no point in making reference
  %% to Pol II.  Other reference dn't even say "only", they may say that
  %% "almost all other protein coding genes" but then have no reference to
  %% what other genes lack poly(A).  I am assuming they are only being
  %% cautious as I can't find information on such genes.


  Canonical core histone transcripts are among the only protein coding messages
  that do not undergo polyadenylation,
  instead carrying a unique stem-loop structure (\tref{tab:typical-histone-differences}).
  They do have a 7--methyl--guanosine cap at the 5' end \citep{MarzluffNatRevGen2008}.

  Canonical histone gene transcription is regulated
  by cell-cycle dependent phosphorylation of the histone-specific NPAT coactivator
  and interaction with the accessory protein FLASH,
  resulting in assembly of histone locus bodies
  coordinating factors responsible for canonical histone transcription and processing
  \citep{MarzluffNatRevGen2008,RattrayMueller2012,Hoefig2014}.
  Variability is observed in canonical histone isoform gene transcription,
  both by analysis of non-polyadenylated transcripts \citep{YangGenomeBiol2011}
  and RNA polymerase II promoter occupancy \citep{Ederveen2011}.

  Overall there is estimated to be a 35 fold increase in mammalian
  canonical histone transcripts during S phase,
  principally as a result of a 10 fold increase in mRNA stabilisation
  via stem-loop dependent mechanisms,
  and a 3--5 fold up-regulation in canonical histone gene transcription \citep{HarrisMCB1991}.

  This post-translation regulation of canonical histone transcripts
  is achieved by a stem-loop encoded in the mRNA 3' untranslated region
  that is recognised by specialised spliceosome--related RNA
  processing and stabilisation complexes.
  The stem-loop is encoded from \StemLoopStart{} to \StemLoopEnd{} bp after the stop codon
  and has a highly conserved sequence and structure
  (\fref{fig:stem-loop-seqlogo} \citep{stem-loop-structure}).
  The RNA stem-loop structure is bound by the Stem-Loop Binding Protein (SLBP)
  which is up-regulated 10--20 fold during S~phase to stabilise
  histone mRNAs \citep{SLBP-regulation}.
  Immediately downstream of the stem-loop a purine-rich Histone Downstream Element (HDE)
  interacts with U7 snRNA to direct efficient 3' end
  processing (\fref{fig:HDE-seqlogo}) \citep{HDE-sequence}.

  \begin{figure}
    \centering
    \subtop[Stem-loop]{
      \includegraphics{seqlogo_stem_loops.pdf}
      \label{fig:stem-loop-seqlogo}
    }
    \hfill
    \subtop[HDE]{
      \includegraphics{seqlogo_HDEs.pdf}
      \label{fig:HDE-seqlogo}
    }
    \caption{%
      Sequence logos for stem-loops and Histone Downstream Elements (HDEs)
      from alignment of all canonical histone mRNA 3' untranslated
      regions (UTRs).
    }
  \end{figure}

  Stabilisation and processing extend the half life of canonical histone mRNAs
  and contribute to increased histone translation efficiency,
  enabling rapid production of canonical histones to package the newly duplicated genome.

  Although the vast bulk of canonical histone transcripts appear to depend on this mechanism,
  \SIrange{1}{5}{\percent} of canonical histone transcripts
  are 3' polyadenylated \citep{YangGenomeBiol2011}
  and some genes have annotations indicating two transcripts
  differing in whether they have stem-loop or polyadenylation
  signals (\tref{tab:curation-anomalies}).
  Histone variant transcripts lack a 3' stem-loop and are
  polyadenylated in the same way as all other protein coding genes.
  The exception is H2AX, which has alternatively processed transcripts
  exhibiting the stem-loop characteristic of canonical histones
  or a poly(A) tail found on other variant histones \citep{HTwoAX-transcripts,our-H2AX-review}.
