\section{The transcripts}
  %% we must say mRNA instead of genes because: 1) we can't defined gene (Pearson.
  %% Genetics: what is a gene?. Nature (2006) vol. 441 http://dx.doi.org/10.1038/441398a )
  %% and 2) it's only for genes transcribed by pol II (the onf for mRNAs)
  Unlike all other eukaryotic mRNAs, histone transcripts are not polyadenylated at
  their 3' end. This process consists in the cleaving of pre-mRNA, and addition of a
  poly(A) tail which promotes the mRNA stability and is involved in its transport
  into the citoplasm \citep{mRNA-end-processing}. The alternative method used by
  histones is the stem-loop, a specialized mRNA secondary.

  \subsection{Stem-loop}
    The stem-loop is encoded shortly after the stop codon, between xxx and yyy base
    pairs, much earlier than the typical polyadenylation signal. The stem-loop sequence
    has been extremely well conserved throughout evolution but it appears that the
    sequence itself is not so important for its function as its actual structure
    \citep{stem-loop-structure}.

    \missingfigure{consensus of human stem-loop.}

    The stem-loop is followed by a purine-rich Histone Downstream Element (HDE)
    downstream of the stop codon. The stem-loop interacts with the Stem-Loop
    Binding Protein (SLBP) to stabilise the mRNA in S~phase \citep{SLBP-regulation}
    while the HDE interacts with U7 snRNA to direct efficient 3 end processing
    \citep{HDE-sequence}.


    This specific machinery for mRNA processing is a large part of increase on
    transcription during the S~phase since their expression is also regulated
    by the cell-cycle \addref and allow for the histone mRNA stability \addref.

    Still, we found some canonical histones with annotated poly(A) tails. These annotations
    are likely to have been done automatically. However, we were also able to identify
    some of them on EST libraries which would suggest that they do exist. Since
    this libraries are performed using oligo-dTs, the histone genes are highly
    expressed, and there's only a few occurrences of them, points that this is
    the exception rather than the rule.

    Variant histones on the other side all have a poly(A) tail with notable
    pseudo-exception of H2AFX which has both a stem-loop and a poly(A) tail,
    and transcribes 2 different mRNAs \citep{HTwoAX-transcripts}.

    In general, metazoan canonical histones
    are not thought to be polyadenylated, although \textit{S.\ cerevisiae} histone transcript stability
    appears to be regulated by polyadenylation instead of RNA structures.

  \subsection{mRNA expression levels}
    %% levels of mRNA of each histone gene in 2-3 different human cell lines (at least one primary cell line)
    Expression of the RD histone genes is highly regulated by the cell cycle (hence the RD name). Probably due
    to the cell high demand for histone proteins during DNA replication, their expression levels rise dramatically
    during this phase of the cell cycle. During the rest of the cell cycle, histone genes are still transcribed
    but with much lower levels.

    However, there is only a 5-fold increase in their transcription rate at S~phase, compared to the other phases
    of cell cycle so regulation acts strongly at the post-transcriptional level\addref.

    To note however, that each gene seems to have different levels of increased transcription, even between genes in the
    same cluster.

    RI histone genes have expression dependent on their needs (testis specific, DNA damage, etc).

    mention chicken histone genes, we can remove half of the set, and the other rise in expression to compensate.


